<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>00_all</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_files/libs/quarto-html/quarto.js"></script>
<script src="00_all_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<link href="00_all_files/libs/table1-1.0/table1_defaults.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">00_all</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="all" class="level1">
<h1>All</h1>
<section id="loading-libraries" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries">Loading libraries</h2>
</section>
</section>
<section id="loading-data" class="level1">
<h1>Loading data</h1>
<section id="loading-libraries-1" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-1">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GEOquery"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="make-_raw-dir" class="level3">
<h3 class="anchored" data-anchor-id="make-_raw-dir">Make _raw dir</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data dirs</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/"</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>raw_dir <span class="ot">&lt;-</span> <span class="st">"../data/_raw/"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(data_dir)) {</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(data_dir)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(raw_dir)) {</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(raw_dir)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-geo-data-using-geoquery" class="level3">
<h3 class="anchored" data-anchor-id="loading-geo-data-using-geoquery">Loading GEO data using GEOquery</h3>
<p><a href="https://warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r/geo/" class="uri">https://warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r/geo/</a></p>
<p>On the GEO database, files are stored as .soft files, with metadata in the beginning and (for the case of “GSE11223”) gene expression data at the end.</p>
<p>The getGEO function from GEOquery obtains queries GEO datasets with the GEO id (if specified). In this case for GSE11223. Else, filename can be supplied to import a downloaded file from the GEO database</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>gse11223 <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO=</span><span class="st">"GSE11223"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gds3268 <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO=</span><span class="st">"GDS3268"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Making raw data files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_rds</span>(gds3268, <span class="at">file=</span><span class="fu">file.path</span>(raw_dir,<span class="st">"/raw_expression_data.rds"</span>))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_rds</span>(gse11223, <span class="at">file=</span><span class="fu">file.path</span> (raw_dir,<span class="st">"/raw_meta_data.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="working-with-the-raw-gse-data" class="level4">
<h4 class="anchored" data-anchor-id="working-with-the-raw-gse-data">Working with the raw “gse” data</h4>
<p>For GSE11223, getGEO pulls the data into a list of length 1. The file GSE11223_series_matrix.txt.gz is extracted into the first (and only) position of that list. We therefore extract the data, getting rid of the parent list:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gse11223 <span class="ot">&lt;-</span> gse11223<span class="sc">$</span>GSE11223_series_matrix.txt.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, the gse11223 variable is a “Large ExpressionSet” of 44.1MB.</p>
</section>
<section id="working-with-the-curated-gds-data" class="level4">
<h4 class="anchored" data-anchor-id="working-with-the-curated-gds-data">Working with the curated “gds” data</h4>
<p>gds3268 is the curated dataset for the project, meaning that the meta data of samples and the expression table can be easily extracted using Meta() and Table() respectively.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gds3268_meta <span class="ot">&lt;-</span> <span class="fu">Meta</span>(gds3268)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gds3268)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gds3268_expression <span class="ot">&lt;-</span> <span class="fu">Table</span>(gds3268)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">slice_sample</span>( gds3268_expression, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   ID_REF      IDENTIFIER GSM282855 GSM282856 GSM282857 GSM282858 GSM282859
1    7389           DGAT2 -0.249220 -0.166110 -0.221690 -0.184670 -0.248640
2   14595            BAG2 -0.785030 -0.871690 -0.871500 -0.826080 -0.754840
3   32667          RPS27A -0.007987 -0.088013 -0.005629  0.011967 -0.000792
4   17170        AW997516  0.005189  0.028165  0.030446  0.046394  0.050534
5   23359          SEPHS1 -0.261840 -0.374030 -0.318340 -0.231840 -0.264040
6   40519     A_23_P68240  0.136390  0.084973  0.105890  0.136460  0.152960
7    6300           ACTG2  1.122400  0.569390  1.134300  0.361430  0.900360
8    3235     A_32_P11673  0.364200  0.305010  0.555470  0.281100  0.367470
9    4753 NegativeControl        NA        NA        NA        NA        NA
10  43731    A_32_P101195  0.321590  0.300990  0.272200  0.082589  0.175560
   GSM282860 GSM282861 GSM282862 GSM282863 GSM282864 GSM282865 GSM282866
1  -0.166390 -0.239430 -0.230980 -0.240460 -0.192920 -0.252100 -0.154220
2  -0.678970 -0.437600 -0.620110 -0.882620 -0.759940 -0.881870 -0.903020
3   0.046408  0.030845 -0.012479 -0.003917  0.048841  0.064003  0.016997
4   0.041824  0.070885  0.061545  0.065646 -0.009442 -0.023695 -0.040375
5  -0.304890 -0.333450 -0.280250 -0.287990 -0.248820 -0.207770 -0.276390
6   0.174760  0.059896  0.104930  0.095086  0.074037  0.131920  0.090805
7   1.070000  1.624400  1.111500  0.657860  0.901800  0.992950  0.636860
8   0.349920  0.537520  0.157880  0.202810  0.522340  0.448560  0.159520
9         NA        NA        NA        NA        NA        NA        NA
10  0.200690  0.275630  0.132040  0.106780  0.362650  0.270570  0.359740
   GSM282867 GSM282868 GSM282869 GSM282870 GSM282871 GSM282872 GSM282904
1  -0.192040 -0.071348 -0.058842 -0.083251 -0.073108 -0.306480 -0.206440
2  -0.771270 -0.753410 -0.782760 -0.578970 -0.491530 -0.844060 -1.279200
3  -0.013903  0.093024 -0.150890 -0.024533  0.028532  0.036600 -0.470020
4   0.078761 -0.018693 -0.073412 -0.086737  0.035414 -0.025185 -0.135530
5  -0.238200 -0.138000 -0.324150 -0.231670 -0.248990 -0.227330 -0.344670
6   0.131550  0.116360  0.097343  0.062171  0.077885  0.127190  0.227390
7   0.728170  1.066300  1.129400  1.022800  1.210600  0.341490  1.074000
8   0.075638  0.082313  0.279470  0.135620  0.080427  0.278920  0.015146
9         NA        NA        NA        NA        NA        NA        NA
10  0.136800  0.230530  0.655770  0.459520  0.358610  0.522820  0.740170
   GSM282910 GSM282913 GSM282915 GSM282921 GSM282927 GSM282873 GSM282874
1  -0.190390 -0.172870 -0.117870 -0.236510 -0.337980 -0.092065 -0.226800
2  -0.998400 -0.930000 -0.991840 -0.953300 -0.906170 -0.887240 -0.707510
3  -0.016846  0.025697 -0.027403  0.029635  0.034291 -0.029449  0.129990
4  -0.007717 -0.035780 -0.015432 -0.000393  0.041769  0.066512  0.084256
5  -0.325510 -0.265750 -0.293520 -0.301100 -0.315680 -0.304350 -0.255960
6   0.211340  0.169170  0.264250  0.163500  0.316320  0.162990  0.157060
7   0.704330  0.981990  0.553340  0.136660  0.953420  1.001400  0.849300
8   0.142870  0.165840  0.430010  0.286680  0.374380 -0.182540 -0.068788
9         NA        NA        NA        NA        NA        NA        NA
10  0.331900  0.347140  0.314040  0.553680  0.467740  0.281110 -0.021059
   GSM282875 GSM282905 GSM282914 GSM282918 GSM282876 GSM282877 GSM282878
1  -0.045379  0.020775 -0.147870 -0.075716 -0.183440 -0.217100 -0.242910
2  -0.603390 -0.677830 -0.863810 -0.900580 -1.100600 -0.963670 -0.907860
3   0.097160  0.033492 -0.161700 -0.123880 -0.228940 -0.044761 -0.083252
4   0.085898  0.031202 -0.091086  0.035361  0.025324  0.030833 -0.010520
5  -0.351040 -0.377410 -0.297650 -0.292830 -0.330820 -0.293240 -0.308400
6   0.180330  0.173870  0.133150  0.110560  0.171120  0.163640  0.200320
7   1.124800  0.538480  1.395300  1.063700  0.667150  0.828860  0.885270
8   0.116390 -0.094913  0.041577 -0.252040  0.127110  0.219300  0.185650
9         NA        NA        NA        NA        NA        NA        NA
10  0.077835  0.496910  0.666770  0.603690  0.227290  0.024023  0.100740
   GSM282879 GSM282880 GSM282881 GSM282882 GSM282883 GSM282884 GSM282885
1  -0.285020 -0.209790 -0.215280 -0.248730 -0.202970 -0.262160 -0.192920
2  -0.861900 -0.804730 -0.903200 -1.149100 -0.961580 -0.880450 -0.857040
3  -0.023085  0.011121 -0.197570 -0.184490 -0.045147 -0.112950 -0.070090
4   0.017466  0.043628  0.000929 -0.040206  0.003337 -0.009527 -0.049515
5  -0.267980 -0.372610 -0.314430 -0.358630 -0.282860 -0.298930 -0.344720
6   0.222680  0.227370  0.150780  0.187520  0.152560  0.258790  0.179910
7   0.884540  1.050100  1.164500  0.454920  0.755970  1.246000  1.279400
8   0.441920  0.436570  0.422600  0.172230  0.185100  0.495080  0.430680
9         NA        NA        NA        NA        NA        NA        NA
10 -0.033273  0.057414  0.108950  0.424520  0.016111  0.359850 -0.006844
   GSM282886 GSM282887 GSM282888 GSM282889 GSM282890 GSM282902 GSM282903
1  -0.344380 -0.203620 -0.366530 -0.273440 -0.270420 -0.191790 -0.156830
2  -1.024100 -1.010100 -0.840690 -0.770070 -0.957130 -0.754070 -0.871720
3  -0.116430 -0.216730 -0.158780 -0.081761 -0.226440  0.038987 -0.138600
4   0.011519 -0.008243 -0.023139  0.056983 -0.133100  0.072824  0.048755
5  -0.350420 -0.370460 -0.323340 -0.378520 -0.367910 -0.343750 -0.249210
6   0.271620  0.255710  0.193670  0.221360  0.219430  0.178050  0.271980
7   0.803870  1.063200  0.536590  1.410200  0.080032  1.010300  1.438900
8   0.573450  0.380150  0.356970  0.310360  0.431540  0.453820  0.147230
9         NA        NA        NA        NA        NA        NA        NA
10  0.146950  0.104930  0.154260 -0.008596  0.295410  0.319560  0.571830
   GSM282907 GSM282909 GSM282912 GSM282920 GSM282924 GSM282891 GSM282892
1  -0.274120 -0.158100 -0.232310 -0.149090 -0.317540 -0.174820 -0.110080
2  -0.948860 -0.962680 -0.923200 -1.082600 -0.839740 -1.144800 -0.914370
3  -0.062925 -0.051991 -0.012358 -0.061666  0.008799 -0.397850 -0.069073
4   0.054552  0.031319  0.038719 -0.068367  0.035490 -0.017774  0.019549
5  -0.367480 -0.262610 -0.322800 -0.260190 -0.286810 -0.296370 -0.331970
6   0.243890  0.165290  0.162830  0.235280  0.262240  0.172700  0.125550
7   0.135580  0.678440  0.203520  0.251730  0.956230 -0.467650  0.944980
8   0.296360  0.104310 -0.016234  0.165310  0.395560  0.126800  0.590800
9         NA        NA        NA        NA        NA        NA        NA
10  0.485920  0.523310  0.351070  0.649620  0.422330  0.624920  0.506780
   GSM282893 GSM282894 GSM282895 GSM282896 GSM282897 GSM282898 GSM282899
1  -0.189900 -0.102300 -0.245110 -0.176210 -0.269160 -0.243390  -0.21349
2  -1.091400 -1.106100 -1.021200 -1.060800 -1.117000 -1.048300  -0.71866
3  -0.068108 -0.292250 -0.024212 -0.047170 -0.013011 -0.072607  -0.13302
4   0.037407  0.016023  0.043434  0.015541 -0.002556  0.032732   0.04421
5  -0.241130 -0.325880 -0.272240 -0.198400 -0.299990 -0.239210  -0.27723
6   0.144660  0.151880  0.127260  0.129070  0.135950  0.145170   0.15164
7   0.456870  0.151630  0.541210  0.651290  0.195990  0.565750   1.44340
8   0.582720  0.550600  0.537940  0.620760  0.727140  0.276640   0.47492
9         NA        NA        NA        NA        NA        NA        NA
10  0.470390  0.642020  0.431050  0.355060  0.330480  0.407740   0.57423
   GSM282900 GSM282901 GSM282906 GSM282908 GSM282911 GSM282916 GSM282919
1  -0.216850  -0.20600 -0.197740  -0.04374 -0.226580 -0.189280 -0.199650
2  -0.762890  -1.06780 -1.064000  -1.25090 -1.062600 -0.947320 -0.956040
3  -0.028509  -0.05903 -0.054124  -0.36230 -0.046184 -0.035713 -0.015096
4  -0.038754   0.00826  0.027865  -0.17027  0.014565 -0.000704  0.001068
5  -0.212330  -0.25429 -0.327180  -0.05190 -0.296480 -0.254370 -0.320660
6   0.083508   0.18196  0.204450   0.25361  0.188680  0.156200  0.201250
7   0.760890   0.09006  1.051000   1.75050  0.752290  0.551880  0.668780
8   0.338470   0.27700  0.056350   0.15985  0.311610  0.290910  0.288520
9         NA        NA        NA        NA        NA        NA        NA
10  0.496070   0.54617  0.637490   1.11580  0.464220  0.555320  0.533520
   GSM282923 GSM282917 GSM282922 GSM282926 GSM282925 GSM282935 GSM282938
1  -0.228950 -0.185070 -0.310150 -0.411060 -0.332980 -0.209420 -0.325040
2  -0.892920 -1.013400 -0.710360 -0.834110 -0.772960 -0.824710 -0.708320
3  -0.129240 -0.009254 -0.006279 -0.055659 -0.012014 -0.025001 -0.001266
4   0.021155  0.039191  0.012268  0.000716  0.029153  0.025154  0.070023
5  -0.242510 -0.361980 -0.266410 -0.257240 -0.310690 -0.272020 -0.307500
6   0.163910  0.272290  0.165760  0.270670  0.237730  0.126900  0.119870
7   1.095600  0.810110  1.217900  1.084000  1.207200  1.008700  0.968240
8   0.099588  0.703680  0.405440  0.385390  0.320350  0.311280  0.466710
9         NA        NA        NA        NA        NA        NA        NA
10  0.522410  0.550570  0.592360  0.635010  0.572010  0.347330  0.464930
   GSM282940 GSM282941 GSM282943 GSM282944 GSM282946 GSM282947 GSM282948
1  -0.209260 -0.243010 -0.267790 -0.258290 -0.277270 -0.319480 -0.228040
2  -0.802320 -0.748830 -0.998130 -0.781850 -0.911060 -0.794700 -0.831300
3  -0.005594 -0.005073 -0.006488 -0.002936 -0.068514  0.084938  0.047965
4  -0.009652  0.002411 -0.023275  0.005416  0.017797 -0.014149 -0.013978
5  -0.217720 -0.266910 -0.372710 -0.273210 -0.272880 -0.234030 -0.268110
6   0.067684  0.178240  0.109520  0.186010  0.114610  0.105510  0.122080
7   0.844080  0.843320  0.847420  1.179800  0.957620  0.867580  1.213500
8   0.013819  0.364570  0.247230  0.159980  0.333340  0.144860  0.359130
9         NA        NA        NA        NA        NA        NA        NA
10  0.128470  0.055735  0.512630  0.316000  0.366050  0.383850  0.377900
   GSM282949 GSM282950 GSM282951 GSM282952 GSM282953 GSM282955 GSM282956
1  -0.146440 -0.119000 -0.293450 -0.281080 -0.251590 -0.158640 -0.154670
2  -0.735250 -0.764080 -0.718130 -0.911080 -0.862270 -0.812040 -0.877160
3   0.038509  0.032105 -0.072161 -0.108820  0.044856  0.064300 -0.068155
4   0.049688 -0.013661  0.031982  0.041703  0.006439  0.047706 -0.148620
5  -0.274310 -0.272180 -0.299230 -0.247590 -0.308620 -0.231860 -0.332570
6   0.116100  0.130880  0.215910  0.165300  0.145600  0.126540  0.167450
7   0.946060  0.842350  1.122800  0.881460  0.786120  1.466100  1.163800
8   0.350230  0.228200  0.141190  0.337350  0.395110        NA  0.010437
9         NA        NA        NA        NA        NA        NA        NA
10  0.357670  0.468430  0.187030  0.202420  0.411230  0.323790  0.723630
   GSM282959 GSM282966 GSM282968 GSM282974 GSM283016 GSM283021 GSM283024
1  -0.317890 -0.222790 -0.212630 -0.258080 -0.073942 -0.142070 -0.144970
2  -0.887570 -0.795370 -0.847920 -0.583950 -0.426140 -0.600460 -1.096500
3   0.149440  0.102110  0.038997  0.106560 -0.103970 -0.256220 -0.213830
4   0.094877 -0.007838 -0.000358 -0.006886 -0.080735 -0.010953 -0.090003
5  -0.280380 -0.318430 -0.268890 -0.137740 -0.403990 -0.295320 -0.285070
6   0.180480  0.250960  0.172010  0.175270  0.197580  0.281630  0.210560
7   0.766640  0.833350  0.684860  0.825670  1.989100  1.663500  0.788480
8   0.313060  0.227730  0.506860  0.011628  0.155150  0.490060  0.480340
9         NA        NA        NA        NA        NA        NA        NA
10 -0.064565  0.418940  0.140200  0.080054  0.674500  0.744610  0.701830
   GSM283041 GSM283043 GSM282957 GSM282958 GSM282960 GSM282971 GSM283015
1  -0.199110 -0.213060 -0.156700 -0.187970 -0.365020 -0.141780  -0.07701
2  -0.773650 -0.894470 -0.584030 -0.730250 -0.962650 -0.470400  -0.94561
3   0.013076  0.063851  0.169000  0.170360  0.089777  0.196240  -0.08482
4   0.018508  0.033529  0.052322  0.126540  0.050285  0.043185  -0.11999
5  -0.394830 -0.329230 -0.317660 -0.287700 -0.196830 -0.287620  -0.30518
6   0.237280  0.186140  0.108600  0.143110  0.091063  0.247550   0.19028
7   1.059200  0.713730  0.686830  0.914920  0.568960  1.013700   1.11560
8   0.149690  0.465280 -0.031748 -0.189350  0.315040 -0.091250  -0.19413
9         NA        NA        NA        NA        NA        NA        NA
10  0.810730  0.371920  0.020255  0.051012 -0.028999 -0.009975   0.69094
   GSM282962 GSM282963 GSM282977 GSM282978 GSM282987 GSM282988 GSM282989
1  -0.252390 -0.351270 -0.240930 -0.196790 -0.195850 -0.191170 -0.171340
2  -0.788010 -1.069000 -0.858080 -0.902570 -0.893360 -1.031700 -0.984470
3  -0.051816  0.014479  0.023335 -0.001679 -0.008878 -0.047015 -0.017004
4   0.033933  0.015716  0.058510 -0.078530  0.048754  0.036246  0.004536
5  -0.217560 -0.322520 -0.315460 -0.351330 -0.229060 -0.235530 -0.298020
6   0.431440  0.208410  0.270920  0.291410  0.137050  0.216890  0.257750
7   0.834010  0.523700  0.860290  0.710340  1.027600  0.684010  0.775010
8   0.138430  0.352450  0.415790  0.383520  0.152800  0.234760  0.513860
9         NA        NA        NA        NA        NA        NA        NA
10 -0.154960 -0.184650 -0.178340 -0.088569 -0.027770 -0.000049 -0.125290
   GSM282990 GSM282991 GSM282992 GSM282993 GSM282994 GSM282995 GSM283020
1  -0.228640 -0.168730 -0.227630 -0.201570 -0.164810 -0.208740 -0.230880
2  -0.848270 -0.928740 -0.881260 -0.675190 -1.054200 -0.707260 -0.849500
3  -0.062476  0.005990 -0.109130 -0.012045 -0.024147 -0.024420 -0.120120
4   0.047675 -0.002915 -0.008062  0.033452  0.002348  0.023498 -0.027383
5  -0.317130 -0.295250 -0.275620 -0.231730 -0.278780 -0.270810 -0.281370
6   0.188470  0.109930  0.282080  0.127080  0.178880  0.206210  0.273100
7   0.674090  0.609140  0.975460  1.419300  0.104530  1.071100  1.093200
8   0.290000  0.350820  0.247810  0.337490  0.323050  0.505290  0.422250
9         NA        NA        NA        NA        NA        NA        NA
10 -0.032226 -0.138580 -0.046235 -0.112140 -0.067571 -0.193330  0.736370
   GSM283023 GSM282931 GSM282939 GSM282981 GSM282983 GSM282985 GSM283000
1  -0.108350 -0.290480 -0.250460 -0.232160 -0.205940 -0.216950 -0.223870
2  -1.003900 -0.861410 -0.769500 -0.849450 -1.008700 -0.960910 -1.081100
3  -0.414820  0.042288 -0.024140 -0.023574  0.033500 -0.009216 -0.069052
4  -0.020259 -0.011952 -0.010827  0.013594 -0.046464  0.031605 -0.095298
5  -0.265400 -0.291100 -0.248260 -0.311390 -0.239910 -0.301170 -0.331170
6   0.252800  0.218400  0.063505  0.257620  0.162690  0.217100  0.120520
7   0.827110  0.602840  0.695610  0.842670  0.505370  0.574630  0.068488
8   0.257930  0.140960  0.346280  0.316960  0.287190  0.301910  0.420960
9         NA        NA        NA        NA        NA        NA        NA
10  0.631050  0.219700  0.117640  0.552150  0.351410  0.367560  0.369790
   GSM283001 GSM283002 GSM283003 GSM283004 GSM283005 GSM283006 GSM283007
1  -0.176720 -0.278750 -0.285730 -0.181090 -0.163370 -0.095513 -0.193470
2  -0.889630 -0.991740 -1.112100 -0.892330 -1.033500 -1.078200 -0.834650
3  -0.095975 -0.101950 -0.135380 -0.033972 -0.166290 -0.157380  0.058351
4  -0.125460  0.043102 -0.030609  0.012150  0.045838 -0.009351  0.010379
5  -0.379620 -0.362510 -0.306470 -0.251870 -0.294760 -0.330220 -0.309900
6   0.227550  0.161990  0.199180  0.154940  0.237160  0.179720  0.155970
7   1.055900  0.715220  0.743090  1.087100  0.818060  0.713220  1.005900
8   0.352240  0.427620  0.798520  0.531900  0.278750  0.382760  0.647630
9         NA        NA        NA        NA        NA        NA        NA
10  0.720940  0.360550  0.589680  0.231410  0.570010  0.642750  0.440620
   GSM283008 GSM283009 GSM283010 GSM283011 GSM283022 GSM283034 GSM283049
1  -0.163420 -0.200420 -0.230400 -0.156650 -0.341800 -0.090373 -0.170870
2  -0.919540 -0.938460 -0.803450 -1.106200 -1.050500 -0.724050 -0.929970
3  -0.051967 -0.039751  0.094456 -0.088368 -0.147040 -0.005001 -0.042035
4  -0.010044  0.009577 -0.011342 -0.071013 -0.055821 -0.020332  0.054276
5  -0.236300 -0.248670 -0.274290 -0.292820 -0.219390 -0.231070 -0.234590
6   0.122830  0.111320  0.174850  0.146940  0.228810  0.087828  0.206370
7   0.957060  0.927440  1.091200  0.682260  0.481630  0.680410  0.707940
8   0.287700  0.719390  0.666970  0.701940  0.447470  0.058074  0.195470
9         NA        NA        NA        NA        NA        NA        NA
10  0.496880  0.476110  0.506310  0.561140  0.673910  0.538440  0.406990
   GSM283051 GSM282929 GSM282933 GSM282936 GSM282937 GSM282942 GSM282945
1  -0.244190 -0.287260 -0.355570 -0.218090 -0.326640 -0.212850 -0.256550
2  -0.777390 -0.550330 -0.712570 -0.445290 -0.890040 -0.613000 -0.838930
3   0.074048  0.154640 -0.190670  0.025910 -0.140800 -0.019036 -0.178250
4  -0.063045  0.025816 -0.080201  0.045622 -0.076559 -0.009758 -0.018447
5  -0.358760 -0.255750 -0.294760 -0.267790 -0.387860 -0.351770 -0.316020
6   0.181610  0.085225  0.256070  0.253800  0.274390  0.136220  0.091653
7   0.013010  1.168500  0.653450  1.150100 -0.070379  1.376400  0.438620
8  -0.109270 -0.091864  0.131220  0.146500  0.079904  0.289760  0.076865
9         NA        NA        NA        NA        NA        NA        NA
10  0.635330 -0.039300  0.717580  0.729920  1.075300  0.416760  0.442170
   GSM282954 GSM282961 GSM282964 GSM282965 GSM282967 GSM282969 GSM282970
1  -0.129840 -0.384160 -0.113840 -0.160490 -0.142380 -0.157540  0.064569
2  -0.784180 -1.047300 -0.725720 -0.989240 -1.131500 -0.768590 -0.408900
3  -0.020251 -0.102640 -0.034094 -0.032551 -0.012906  0.095065  0.003346
4   0.052422  0.061246 -0.019848  0.008443 -0.016301  0.008566 -0.005063
5  -0.310010 -0.364400 -0.311390 -0.196990 -0.289960 -0.286340 -0.317540
6   0.143810  0.196870  0.176170  0.122820  0.185330  0.194080  0.196880
7   0.677300 -0.238790  0.472100  0.780020 -0.185960  0.490580  0.291830
8         NA  0.345460 -0.188920  0.240890  0.284350  0.014733 -0.072240
9         NA        NA        NA        NA        NA        NA        NA
10  0.398130  0.103860  0.485080  0.435570  0.392990  0.627590  0.695020
   GSM282972 GSM282973 GSM282975 GSM282996 GSM282999 GSM283014 GSM283019
1  -0.181300 -0.171870 -0.375010 -0.149710 -0.231130 -0.049022 -0.238580
2  -0.534600 -1.089800 -0.656740 -0.556090 -0.896990 -0.367170 -1.053200
3   0.085401 -0.085649  0.102510 -0.072599 -0.116760  0.011295 -0.078365
4   0.049633  0.009296  0.026680 -0.092801 -0.067475 -0.092124  0.007252
5  -0.287070 -0.311430 -0.283430 -0.318890 -0.398220 -0.371070 -0.334670
6   0.125700  0.154140  0.147440  0.151270  0.310190  0.218310  0.209680
7   1.477300  0.376390  1.247600  1.802000  0.831270  1.935700  0.733680
8   0.200900  0.478700  0.081082  0.060070  0.075796  0.207070  0.578270
9         NA        NA        NA        NA        NA        NA        NA
10  0.101490  0.155460  0.017257  0.856440  0.616180  0.821360  0.529840
   GSM283026 GSM283029 GSM283030 GSM283033 GSM283035 GSM283036 GSM283038
1  -0.210900 -0.249260 -0.261440 -0.101710 -0.305070 -0.296820 -0.282440
2  -0.737190 -0.994030 -0.715510 -0.702840 -0.545060 -0.682870 -0.644160
3  -0.005846 -0.254170  0.002672 -0.024479  0.008934  0.031228  0.004730
4  -0.036895 -0.025426  0.047019 -0.016075  0.026194  0.074550  0.025153
5  -0.311780 -0.400300 -0.347280 -0.268990 -0.267620 -0.338770 -0.254900
6   0.124350  0.219340  0.202290  0.071141  0.236780  0.262580  0.146280
7   0.984740  0.705620  0.878750  0.713910  1.093400  1.292500  1.074100
8   0.205950  0.127940  0.258010  0.188260  0.146430  0.524820  0.353130
9         NA        NA        NA        NA        NA        NA        NA
10  0.477050  0.774960  0.713490  0.505000  0.627310  0.494330  0.530510
   GSM283046 GSM283050 GSM283053 GSM283055 GSM283056 GSM282928 GSM282930
1  -0.281740 -0.151380 -0.229050 -0.264980 -0.162240 -0.302690 -0.240870
2  -0.715720 -0.981690 -0.728510 -0.703890 -0.389560 -0.639400 -0.884280
3   0.013029 -0.223790 -0.004336 -0.035867  0.043615  0.009902 -0.038986
4   0.047780 -0.062596 -0.001663  0.040923 -0.013077  0.030128 -0.062346
5  -0.315140 -0.373190 -0.306260 -0.377800 -0.294200 -0.354530 -0.340730
6   0.156220  0.246770  0.102130  0.308620  0.131190  0.214760  0.308740
7   1.023700 -0.319580  0.637540  1.232600  1.342600  1.095200  0.931290
8   0.229760  0.282120  0.468390  0.549060  0.391890  0.298980  0.060706
9         NA        NA        NA        NA        NA        NA        NA
10  0.613860  0.889950  0.518710  0.489080  0.410150  0.200250 -0.034090
   GSM282932 GSM282934 GSM282976 GSM282979 GSM282998 GSM283013 GSM283017
1  -0.260250 -0.179590 -0.289400 -0.235920 -0.083836 -0.045361 -0.322230
2  -1.109500 -0.752830 -1.057000 -0.726850 -0.754780 -0.718470 -0.583970
3  -0.138760 -0.271310 -0.009267 -0.138540  0.019205 -0.363850  0.037505
4   0.015028 -0.155350  0.036961  0.009857 -0.111660 -0.106900  0.003732
5  -0.337200 -0.357240 -0.299680 -0.393990 -0.398150 -0.358580 -0.251270
6   0.289750        NA  0.138800  0.107520  0.219370  0.168430  0.253150
7   0.853430  1.245200  0.559350  1.204800  0.412630  1.489700  1.526600
8   0.460380 -0.007406  0.016081  0.061962 -0.066877  0.152370  0.333840
9         NA        NA        NA        NA        NA        NA        NA
10  0.637650        NA  0.006574 -0.081191  0.680870  0.832680  0.546340
   GSM283018 GSM283025 GSM283028 GSM283032 GSM283037 GSM283040 GSM283042
1  -0.270340 -0.188610 -0.292480 -0.156110 -0.329990 -0.309220 -0.243740
2  -0.918270 -0.601230 -0.636820 -0.688010 -0.578890 -0.867680 -1.021000
3   0.023965  0.084133  0.050947 -0.082040  0.027247 -0.001548  0.039215
4   0.014508  0.024049  0.013728  0.020567 -0.002105  0.009425 -0.014455
5  -0.361810 -0.321500 -0.334130 -0.291040 -0.242960 -0.328360 -0.293460
6   0.210870  0.170420  0.152660  0.070438  0.318640  0.231380  0.161830
7   0.435330  0.503480  0.883320  1.094500  1.007000  0.151170  0.027048
8   0.032736  0.118250  0.357730  0.244440  0.238800  0.201470  0.555170
9         NA        NA        NA        NA        NA        NA        NA
10  0.459160  0.501770  0.506120  0.548800  0.475080  0.639240  0.403720
   GSM283045 GSM283048 GSM283052 GSM283054 GSM282980 GSM282982 GSM282984
1  -0.231990 -0.275550 -0.245810 -0.212040 -0.141390 -0.295450 -0.278540
2  -0.618330 -0.764370 -0.872030 -0.288080 -1.024100 -0.880950 -1.024500
3   0.081337  0.036586 -0.058997  0.040285 -0.041091 -0.043189 -0.025871
4   0.045720  0.020668  0.008277 -0.003420 -0.070365  0.048409 -0.022716
5  -0.315230 -0.309340 -0.327310 -0.284310 -0.304880 -0.315110 -0.304260
6   0.231420  0.155220  0.154430  0.092766  0.131350  0.239190  0.223980
7   1.029400  1.204100  0.409790  1.432400  0.033137  1.121500  0.594760
8   0.197140  0.496160  0.364840  0.480270  0.303340  0.290060  0.179480
9         NA        NA        NA        NA        NA        NA        NA
10  0.557460  0.526590  0.594710  0.453860  0.517390  0.581090  0.528700
   GSM282986 GSM282997 GSM283012 GSM283027 GSM283031 GSM283039 GSM283044
1   0.025289 -0.087941 -0.077239 -0.263760 -0.231530 -0.290100 -0.272940
2  -0.880730 -1.032500 -1.152200 -0.823650 -0.883270 -0.862600 -0.927060
3   0.026212 -0.297080 -0.256670  0.009311 -0.170030 -0.023533  0.042248
4  -0.116340 -0.240420 -0.137970 -0.006633  0.013334  0.005568 -0.004131
5  -0.302870 -0.416390 -0.304680 -0.289800 -0.268690 -0.298310 -0.290500
6   0.261840  0.180630  0.205960  0.177600  0.148890  0.150660  0.133940
7   0.622530  0.702770 -0.503770  1.045500  0.468300  0.877880  0.582370
8  -0.032267  0.087972  0.245760  0.389350  0.433770  0.200250  0.282840
9         NA        NA        NA        NA        NA        NA        NA
10  0.592540  0.878180  0.790010  0.382320  0.555580  0.690220  0.444460
   GSM283047
1  -0.255370
2  -0.890910
3   0.057266
4   0.035051
5  -0.271500
6   0.207250
7   0.773840
8   0.347110
9         NA
10  0.526850</code></pre>
</div>
</div>
<p><strong>Save gse11223 &amp; gds3268 objects in _raw/</strong></p>
<p>underlying tables can be extracted from each when they’re loaded as rds files, this ensures collectively of the two pulls when working with the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_rds</span>(gds3268, <span class="at">file=</span><span class="fu">file.path</span>(data_dir,<span class="st">"01_load_expression_data.rds"</span>))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_rds</span>(gse11223, <span class="at">file=</span><span class="fu">file.path</span> (data_dir,<span class="st">"01_load_meta_data.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>_raw/ will indeed only contain raw data pulls (apart from the disturbing list extraction for gse11223…)</p>
<p>Cleaning environment at the end of the file</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
</section>
<section id="cleaning-data" class="level1">
<h1>Cleaning data</h1>
<section id="loading-libraries-2" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-2">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Clean _raw/ data to data/</p>
<p>Converts each of the following to a tibble dataframe in the tidy format</p>
<ul>
<li><p>the expression GEODataTable from GDS in _raw/</p></li>
<li><p>the metadata from GSE</p>
<ul>
<li><p>phenoData$data - metadata concerning samples (patients)</p></li>
<li><p>featureData$data - metadata concerning genes</p></li>
</ul></li>
</ul>
<p><strong>Setup/Initialization</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>raw_dir <span class="ot">&lt;-</span> <span class="st">"../data/_raw/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="geodatatable-from-rawraw_expression_data.rds" class="level2">
<h2 class="anchored" data-anchor-id="geodatatable-from-rawraw_expression_data.rds">GEODataTable from raw/raw_expression_data.rds</h2>
<section id="loading-in-the-data" class="level4">
<h4 class="anchored" data-anchor-id="loading-in-the-data">Loading in the data</h4>
<p>Expression data of patients with different genes. (patient) Samples are stored as columns and gene names as rows, which we would like to flip, to align with tidy standards. But first we load the data from _raw/</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>exprs_geodf <span class="ot">&lt;-</span> <span class="fu">Table</span>(<span class="fu">read_rds</span>(<span class="fu">file.path</span>(data_dir,<span class="st">"01_load_expression_data.rds"</span>)))</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>exprs_geodf <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(exprs_geodf)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>exprs_geodf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 44,290 × 204
   ID_REF IDENTIFIER GSM282855 GSM282856 GSM282857 GSM282858 GSM282859 GSM282860
    &lt;int&gt; &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1      1 BrightCor…   NA        NA        NA        NA        NA        NA     
 2      2 NegativeC…   NA        NA        NA        NA        NA        NA     
 3      3 A_23_P803…    0.0387   -0.109    -0.0443   -0.0334   -0.0436    0.0268
 4      4 APBA2        -0.133    -0.109    -0.136    -0.119    -0.164    -0.185 
 5      5 MAP3K6       -0.0422    0.0422    0.0180    0.0635    0.0720    0.0587
 6      6 ZNF121       -0.212    -0.240    -0.339    -0.314    -0.436    -0.0504
 7      7 Pro25G       NA        NA        NA        NA        NA        NA     
 8      8 RET          NA        -0.115    -0.442    -0.250    -0.440    -0.467 
 9      9 BX094364      0.0346    0.0149   -0.0541   -0.0582   -0.0832   -0.0951
10     10 C12orf56      0.0286   -0.0422    0.0311   -0.0577   -0.0826   -0.0948
# ℹ 44,280 more rows
# ℹ 196 more variables: GSM282861 &lt;dbl&gt;, GSM282862 &lt;dbl&gt;, GSM282863 &lt;dbl&gt;,
#   GSM282864 &lt;dbl&gt;, GSM282865 &lt;dbl&gt;, GSM282866 &lt;dbl&gt;, GSM282867 &lt;dbl&gt;,
#   GSM282868 &lt;dbl&gt;, GSM282869 &lt;dbl&gt;, GSM282870 &lt;dbl&gt;, GSM282871 &lt;dbl&gt;,
#   GSM282872 &lt;dbl&gt;, GSM282904 &lt;dbl&gt;, GSM282910 &lt;dbl&gt;, GSM282913 &lt;dbl&gt;,
#   GSM282915 &lt;dbl&gt;, GSM282921 &lt;dbl&gt;, GSM282927 &lt;dbl&gt;, GSM282873 &lt;dbl&gt;,
#   GSM282874 &lt;dbl&gt;, GSM282875 &lt;dbl&gt;, GSM282905 &lt;dbl&gt;, GSM282914 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we can treat it as a normal dataframe to try to clean it to a tidy dataframe.</p>
</section>
<section id="removing-empty-rows" class="level4">
<h4 class="anchored" data-anchor-id="removing-empty-rows">Removing empty rows</h4>
<p>All 204 columns are (patient) Samples, and of type &lt;dbl&gt; with NA values. With this data we would like to keep the random NA values that appear throughout the dataset as they clearly represent no data. Substituting it with zeros would indicate that there is no expression of that gene in that sample, which is not necessarily the case (most likely not). Although, if an entire row is made of zeros, we would like to exclude that gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>fully_na <span class="ot">&lt;-</span> exprs_geodf <span class="sc">|&gt;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">if_all</span>(<span class="fu">starts_with</span>(<span class="st">"GSM"</span>), is.na)) <span class="sc">|&gt;</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">"ID_REF"</span>) </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">count</span>(fully_na)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 1
      n
  &lt;int&gt;
1  3299</code></pre>
</div>
</div>
<p>There are 3299 identifiers with completely empty rows in the expression dataset, given in the list “fully_na”. We cannot use the gene column name: “IDENTIFIER”, as it contains duplicates. Lets exclude fully_na indexes from the dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> exprs_geodf <span class="sc">|&gt;</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(fully_na, <span class="at">by =</span> <span class="st">"ID_REF"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>exprs_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 40,991 × 204
   ID_REF IDENTIFIER GSM282855 GSM282856 GSM282857 GSM282858 GSM282859 GSM282860
    &lt;int&gt; &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1      3 A_23_P803…    0.0387   -0.109    -0.0443   -0.0334   -0.0436    0.0268
 2      4 APBA2        -0.133    -0.109    -0.136    -0.119    -0.164    -0.185 
 3      5 MAP3K6       -0.0422    0.0422    0.0180    0.0635    0.0720    0.0587
 4      6 ZNF121       -0.212    -0.240    -0.339    -0.314    -0.436    -0.0504
 5      8 RET          NA        -0.115    -0.442    -0.250    -0.440    -0.467 
 6      9 BX094364      0.0346    0.0149   -0.0541   -0.0582   -0.0832   -0.0951
 7     10 C12orf56      0.0286   -0.0422    0.0311   -0.0577   -0.0826   -0.0948
 8     12 MBD2          0.151     0.211     0.196     0.216     0.235     0.220 
 9     13 LIFR-AS1     -0.268    -0.479    -0.323    -0.343    -0.459    -0.434 
10     15 CMYA5        NA        -0.0329   -0.275     0.0600   -0.341    -0.289 
# ℹ 40,981 more rows
# ℹ 196 more variables: GSM282861 &lt;dbl&gt;, GSM282862 &lt;dbl&gt;, GSM282863 &lt;dbl&gt;,
#   GSM282864 &lt;dbl&gt;, GSM282865 &lt;dbl&gt;, GSM282866 &lt;dbl&gt;, GSM282867 &lt;dbl&gt;,
#   GSM282868 &lt;dbl&gt;, GSM282869 &lt;dbl&gt;, GSM282870 &lt;dbl&gt;, GSM282871 &lt;dbl&gt;,
#   GSM282872 &lt;dbl&gt;, GSM282904 &lt;dbl&gt;, GSM282910 &lt;dbl&gt;, GSM282913 &lt;dbl&gt;,
#   GSM282915 &lt;dbl&gt;, GSM282921 &lt;dbl&gt;, GSM282927 &lt;dbl&gt;, GSM282873 &lt;dbl&gt;,
#   GSM282874 &lt;dbl&gt;, GSM282875 &lt;dbl&gt;, GSM282905 &lt;dbl&gt;, GSM282914 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>With our little sanity check below, we can confirm that 3299 rows has been filtered from the data, matching the number of empty rows (fully_na)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(exprs_geodf)[[<span class="dv">1</span>]] <span class="sc">-</span> <span class="fu">dim</span>(exprs_df)[[<span class="dv">1</span>]] <span class="sc">-</span> <span class="fu">dim</span>(fully_na)[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
<p>We could choose to reset ID_REF values and keep them as the index of the frame (essentially removing them), but that would obscure us from combining the expression dataframe with any potential meta data later on. We therefore instead denote the ID_REF column as lookup keys; since they’re unique.</p>
</section>
<section id="transposing-the-dataframe" class="level4">
<h4 class="anchored" data-anchor-id="transposing-the-dataframe">Transposing the dataframe</h4>
<p>As according to tidy principles, we need variables (gene names) as columns and observations (samples) as rows, which is currently flipped. We therefore need to transpose the dataframe.</p>
<p>We can use pivot_longer to concatenate columns and values except <em>ID_REF</em>, and then pivot_wider to unpack the values along with <em>ID_REF</em>, leaving sample names as row names. Although, currently we have both <em>ID_REF</em> and <em>IDENTIFIER</em> functioning as column headers. We choose to exclude the IDENTIFIER column from the table, and combine it with future metadata tables. As long as the ID_REF is present, we can lookup any meta data there might be for either variables and observations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>identifier_exprs <span class="ot">&lt;-</span> exprs_df <span class="sc">|&gt;</span>  <span class="co"># for later use</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID_REF, IDENTIFIER)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> exprs_df <span class="sc">|&gt;</span> </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>IDENTIFIER) <span class="sc">|&gt;</span> </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Packing column names to one (except ID_REF)</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>ID_REF,          </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"geo_accession"</span>, </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"Value"</span>  </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Unpacking ID_REF as columns</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> ID_REF,</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> Value</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>exprs_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 202 × 40,992
   geo_accession      `3`     `4`     `5`     `6`     `8`     `9`    `10`  `12`
   &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
 1 GSM282855      0.0387  -0.133  -0.0422 -0.212  NA       0.0346  0.0286 0.151
 2 GSM282856     -0.109   -0.109   0.0422 -0.240  -0.115   0.0149 -0.0422 0.211
 3 GSM282857     -0.0443  -0.136   0.0180 -0.339  -0.442  -0.0541  0.0311 0.196
 4 GSM282858     -0.0334  -0.119   0.0635 -0.314  -0.250  -0.0582 -0.0577 0.216
 5 GSM282859     -0.0436  -0.164   0.0720 -0.436  -0.440  -0.0832 -0.0826 0.235
 6 GSM282860      0.0268  -0.185   0.0587 -0.0504 -0.467  -0.0951 -0.0948 0.220
 7 GSM282861     -0.0448  -0.0768  0.292  -0.318  -0.141   0.0369  0.0377 0.190
 8 GSM282862     -0.0588  -0.184   0.101  -0.286  -0.113  -0.0992 -0.0966 0.275
 9 GSM282863     -0.0428  -0.108   0.125  -0.0161 -0.430   0.0272 -0.0273 0.281
10 GSM282864     -0.00537 -0.186   0.143  -0.274  -0.0461  0.0356  0.0444 0.108
# ℹ 192 more rows
# ℹ 40,983 more variables: `13` &lt;dbl&gt;, `15` &lt;dbl&gt;, `16` &lt;dbl&gt;, `17` &lt;dbl&gt;,
#   `18` &lt;dbl&gt;, `19` &lt;dbl&gt;, `22` &lt;dbl&gt;, `24` &lt;dbl&gt;, `25` &lt;dbl&gt;, `26` &lt;dbl&gt;,
#   `27` &lt;dbl&gt;, `29` &lt;dbl&gt;, `30` &lt;dbl&gt;, `31` &lt;dbl&gt;, `32` &lt;dbl&gt;, `33` &lt;dbl&gt;,
#   `34` &lt;dbl&gt;, `36` &lt;dbl&gt;, `37` &lt;dbl&gt;, `38` &lt;dbl&gt;, `39` &lt;dbl&gt;, `40` &lt;dbl&gt;,
#   `41` &lt;dbl&gt;, `43` &lt;dbl&gt;, `45` &lt;dbl&gt;, `46` &lt;dbl&gt;, `47` &lt;dbl&gt;, `48` &lt;dbl&gt;,
#   `50` &lt;dbl&gt;, `51` &lt;dbl&gt;, `52` &lt;dbl&gt;, `53` &lt;dbl&gt;, `54` &lt;dbl&gt;, `55` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we see the data transposed with 202 rows and 40,992 columns, as opposed to the opposite before.</p>
</section>
</section>
<section id="removing-low-variance-genes" class="level2">
<h2 class="anchored" data-anchor-id="removing-low-variance-genes">Removing low variance genes</h2>
<p>To lighten the load on the server we also removed low variance genes, also if there is little to no variance the genes are probably not important.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>gene_var <span class="ot">&lt;-</span> exprs_df <span class="sc">|&gt;</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># drop the sample ID column, keep only expression values</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>geo_accession) <span class="sc">|&gt;</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span> <span class="fu">var</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols      =</span> <span class="fu">everything</span>(),</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to  =</span> <span class="st">"gene"</span>,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"variance"</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_var, <span class="fu">aes</span>(<span class="at">x =</span> variance)) <span class="sc">+</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 9 rows containing non-finite outside the scale range
(`stat_bin()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-30-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"variance_distribution_across_genes.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 9 rows containing non-finite outside the scale range
(`stat_bin()`).</code></pre>
</div>
</div>
<p>We decide to cutoff genes with a variance below 0.05, and have 2909 genes remaining</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>genes_keep <span class="ot">&lt;-</span> gene_var <span class="sc">|&gt;</span> </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(variance <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>exprs_filtered <span class="ot">&lt;-</span> exprs_df <span class="sc">|&gt;</span> </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geo_accession, <span class="fu">all_of</span>(genes_keep))</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(genes_keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2909</code></pre>
</div>
</div>
</section>
<section id="phenodatadata-from-_rawgse11223.rds" class="level2">
<h2 class="anchored" data-anchor-id="phenodatadata-from-_rawgse11223.rds">phenoData$data from _raw/gse11223.rds</h2>
<p>Metadata concerning samples (patients), extracting the nested data frame phenodata, while removing</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>gse11223 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(data_dir,<span class="st">"01_load_meta_data.rds"</span>)) </span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> <span class="fu">pData</span>(<span class="fu">phenoData</span>(gse11223[<span class="dv">1</span>]))</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>phenodata <span class="sc">|&gt;</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span> <span class="fu">n_distinct</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))) <span class="sc">|&gt;</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">everything</span>(),</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"col"</span>,</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"n_levels"</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_levels <span class="sc">&lt;=</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 34 × 2
   col                  n_levels
   &lt;chr&gt;                   &lt;int&gt;
 1 status                      1
 2 submission_date             1
 3 last_update_date            1
 4 type                        1
 5 channel_count               1
 6 organism_ch1                1
 7 molecule_ch1                1
 8 extract_protocol_ch1        1
 9 label_ch1                   1
10 label_protocol_ch1          1
# ℹ 24 more rows</code></pre>
</div>
</div>
<p>Once we spotted the “one type columns” we drop them. Then we check the remaining ones.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata <span class="sc">|&gt;</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">n_distinct</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">1</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phenodata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "title"                         "geo_accession"                
  [3] "source_name_ch1"               "characteristics_ch1"          
  [5] "characteristics_ch1.1"         "characteristics_ch1.2"        
  [7] "characteristics_ch1.3"         "characteristics_ch1.4"        
  [9] "characteristics_ch1.5"         "characteristics_ch1.6"        
 [11] "characteristics_ch1.7"         "characteristics_ch1.8"        
 [13] "characteristics_ch1.9"         "characteristics_ch1.10"       
 [15] "characteristics_ch1.11"        "characteristics_ch1.12"       
 [17] "characteristics_ch1.13"        "characteristics_ch1.14"       
 [19] "characteristics_ch1.15"        "characteristics_ch1.16"       
 [21] "characteristics_ch1.17"        "characteristics_ch1.18"       
 [23] "characteristics_ch1.19"        "characteristics_ch1.20"       
 [25] "characteristics_ch1.21"        "characteristics_ch1.22"       
 [27] "characteristics_ch1.23"        "characteristics_ch1.24"       
 [29] "characteristics_ch1.25"        "characteristics_ch1.26"       
 [31] "characteristics_ch1.27"        "characteristics_ch1.28"       
 [33] "characteristics_ch1.29"        "characteristics_ch1.30"       
 [35] "characteristics_ch1.31"        "characteristics_ch1.32"       
 [37] "characteristics_ch1.33"        "characteristics_ch1.34"       
 [39] "characteristics_ch1.35"        "characteristics_ch1.36"       
 [41] "characteristics_ch1.37"        "characteristics_ch1.38"       
 [43] "characteristics_ch1.39"        "characteristics_ch1.40"       
 [45] "characteristics_ch1.41"        "characteristics_ch1.42"       
 [47] "characteristics_ch1.43"        "characteristics_ch1.44"       
 [49] "characteristics_ch1.45"        "characteristics_ch1.46"       
 [51] "characteristics_ch1.47"        "characteristics_ch1.48"       
 [53] "characteristics_ch1.49"        "characteristics_ch1.50"       
 [55] "characteristics_ch1.51"        "characteristics_ch1.52"       
 [57] "characteristics_ch1.53"        "characteristics_ch1.54"       
 [59] "description"                   "supplementary_file"           
 [61] "relation"                      "relation.1"                   
 [63] "albumin:ch1"                   "anatomic_location:ch1"        
 [65] "aza at present:ch1"            "aza tolerant:ch1"             
 [67] "aza treated:ch1"               "birth date:ch1"               
 [69] "blood obtained:ch1"            "calprotectin:ch1"             
 [71] "casenote review date:ch1"      "category:ch1"                 
 [73] "crp:ch1"                       "current medication:ch1"       
 [75] "diagnosis date:ch1"            "diagnosis hep flex:ch1"       
 [77] "diagnosis recto sigmoid:ch1"   "diagnosis rectum:ch1"         
 [79] "diagnosis splen flex:ch1"      "diagnosis total:ch1"          
 [81] "disease:ch1"                   "dlg5 133a:ch1"                
 [83] "esr:ch1"                       "ethnicity:ch1"                
 [85] "family history:ch1"            "followup hep flex:ch1"        
 [87] "followup recto sigmoid:ch1"    "followup rectum:ch1"          
 [89] "followup splen flex:ch1"       "followup total:ch1"           
 [91] "hb:ch1"                        "ibd affected relatives:ch1"   
 [93] "igr2096:ch1"                   "igr2198:ch1"                  
 [95] "igr2230:ch1"                   "indictation for procedure:ch1"
 [97] "inflammation_status:ch1"       "joint problems:ch1"           
 [99] "last followup date:ch1"        "neutrophils:ch1"              
[101] "nod 1007 fs:ch1"               "nod 702:ch1"                  
[103] "nod 908:ch1"                   "octn1:ch1"                    
[105] "octn2:ch1"                     "other illnesses:ch1"          
[107] "patient:ch1"                   "procedure date:ch1"           
[109] "run_date:ch1"                  "smoking amount:ch1"           
[111] "smoking start date:ch1"        "smoking status:ch1"           
[113] "smoking stop date:ch1"         "symptoms onset date:ch1"      
[115] "uc flare up:ch1"               "ucss:ch1"                     
[117] "wcc:ch1"                      </code></pre>
</div>
</div>
<p>Then we remove the duplicate columns and rename them appropriately</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata <span class="sc">|&gt;</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Convert all columns with "characteristics_ch1" to long format</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"characteristics_ch1"</span>),</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"old_col"</span>,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Split</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">col  =</span> value,</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"prefix"</span>, <span class="st">"content"</span>),</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep  =</span> <span class="st">":"</span>,  <span class="co"># split at the first colon</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"right"</span>,   <span class="co"># if content is missing, fill with NA</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Clean the prefix</span></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">prefix =</span> <span class="fu">str_replace_all</span>(prefix, <span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">"_"</span>),</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">content =</span> <span class="fu">str_trim</span>(content)</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>old_col) <span class="sc">|&gt;</span></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Convert back to wide</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> prefix,</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> content</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata <span class="sc">|&gt;</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">ends_with</span>(<span class="st">":ch1"</span>),</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ends_with</span>(<span class="st">":ch1"</span>)</span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phenodata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "title"                         "geo_accession"                
  [3] "source_name_ch1"               "description"                  
  [5] "supplementary_file"            "relation"                     
  [7] "relation.1"                    "patient"                      
  [9] "current_medication"            "birth_date"                   
 [11] "ethnicity"                     "symptoms_onset_date"          
 [13] "diagnosis_date"                "joint_problems"               
 [15] "uc_flare_up"                   "family_history"               
 [17] "aza_treated"                   "aza_tolerant"                 
 [19] "aza_at_present"                "procedure_date"               
 [21] "indictation_for_procedure"     "ucss"                         
 [23] "calprotectin"                  "esr"                          
 [25] "crp"                           "hb"                           
 [27] "wcc"                           "neutrophils"                  
 [29] "albumin"                       "blood_obtained"               
 [31] "ibd_affected_relatives"        "igr2096"                      
 [33] "igr2198"                       "igr2230"                      
 [35] "octn1"                         "octn2"                        
 [37] "nod_702"                       "nod_908"                      
 [39] "dlg5_133a"                     "nod_1007_fs"                  
 [41] "diagnosis_rectum"              "diagnosis_recto_sigmoid"      
 [43] "diagnosis_splen_flex"          "diagnosis_hep_flex"           
 [45] "diagnosis_total"               "last_followup_date"           
 [47] "followup_rectum"               "followup_recto_sigmoid"       
 [49] "followup_splen_flex"           "followup_hep_flex"            
 [51] "followup_total"                "casenote_review_date"         
 [53] "category"                      "smoking_status"               
 [55] "smoking_start_date"            "smoking_stop_date"            
 [57] "smoking_amount"                "other_illnesses"              
 [59] "disease"                       "run_date"                     
 [61] "anatomic_location"             "inflammation_status"          
 [63] "albumin:ch1"                   "anatomic_location:ch1"        
 [65] "aza at present:ch1"            "aza tolerant:ch1"             
 [67] "aza treated:ch1"               "birth date:ch1"               
 [69] "blood obtained:ch1"            "calprotectin:ch1"             
 [71] "casenote review date:ch1"      "category:ch1"                 
 [73] "crp:ch1"                       "current medication:ch1"       
 [75] "diagnosis date:ch1"            "diagnosis hep flex:ch1"       
 [77] "diagnosis recto sigmoid:ch1"   "diagnosis rectum:ch1"         
 [79] "diagnosis splen flex:ch1"      "diagnosis total:ch1"          
 [81] "disease:ch1"                   "dlg5 133a:ch1"                
 [83] "esr:ch1"                       "ethnicity:ch1"                
 [85] "family history:ch1"            "followup hep flex:ch1"        
 [87] "followup recto sigmoid:ch1"    "followup rectum:ch1"          
 [89] "followup splen flex:ch1"       "followup total:ch1"           
 [91] "hb:ch1"                        "ibd affected relatives:ch1"   
 [93] "igr2096:ch1"                   "igr2198:ch1"                  
 [95] "igr2230:ch1"                   "indictation for procedure:ch1"
 [97] "inflammation_status:ch1"       "joint problems:ch1"           
 [99] "last followup date:ch1"        "neutrophils:ch1"              
[101] "nod 1007 fs:ch1"               "nod 702:ch1"                  
[103] "nod 908:ch1"                   "octn1:ch1"                    
[105] "octn2:ch1"                     "other illnesses:ch1"          
[107] "patient:ch1"                   "procedure date:ch1"           
[109] "run_date:ch1"                  "smoking amount:ch1"           
[111] "smoking start date:ch1"        "smoking status:ch1"           
[113] "smoking stop date:ch1"         "symptoms onset date:ch1"      
[115] "uc flare up:ch1"               "ucss:ch1"                     
[117] "wcc:ch1"                      </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata[, <span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">as.list</span>(phenodata))]</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phenodata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "title"                     "geo_accession"            
 [3] "source_name_ch1"           "supplementary_file"       
 [5] "relation"                  "patient"                  
 [7] "current_medication"        "birth_date"               
 [9] "ethnicity"                 "symptoms_onset_date"      
[11] "diagnosis_date"            "joint_problems"           
[13] "uc_flare_up"               "family_history"           
[15] "aza_treated"               "aza_tolerant"             
[17] "aza_at_present"            "procedure_date"           
[19] "indictation_for_procedure" "ucss"                     
[21] "calprotectin"              "esr"                      
[23] "crp"                       "hb"                       
[25] "wcc"                       "neutrophils"              
[27] "albumin"                   "blood_obtained"           
[29] "ibd_affected_relatives"    "igr2096"                  
[31] "igr2198"                   "igr2230"                  
[33] "octn1"                     "octn2"                    
[35] "nod_702"                   "nod_908"                  
[37] "nod_1007_fs"               "diagnosis_rectum"         
[39] "diagnosis_recto_sigmoid"   "diagnosis_splen_flex"     
[41] "diagnosis_hep_flex"        "diagnosis_total"          
[43] "last_followup_date"        "followup_rectum"          
[45] "followup_recto_sigmoid"    "followup_splen_flex"      
[47] "followup_hep_flex"         "followup_total"           
[49] "casenote_review_date"      "category"                 
[51] "smoking_status"            "smoking_start_date"       
[53] "smoking_stop_date"         "smoking_amount"           
[55] "other_illnesses"           "disease"                  
[57] "run_date"                  "anatomic_location"        
[59] "inflammation_status"      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>phenodata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 202 × 59
   title       geo_accession source_name_ch1 supplementary_file relation patient
   &lt;chr&gt;       &lt;chr&gt;         &lt;chr&gt;           &lt;chr&gt;              &lt;chr&gt;    &lt;chr&gt;  
 1 5192 Norma… GSM282855     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 101    
 2 5193 Norma… GSM282856     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 102    
 3 5194 Norma… GSM282857     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 103    
 4 5195 Norma… GSM282858     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 105    
 5 5196 Norma… GSM282859     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 105    
 6 5197 Norma… GSM282860     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 106    
 7 5198 Norma… GSM282861     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 108    
 8 5199 Norma… GSM282862     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 104    
 9 5200 Norma… GSM282863     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 107    
10 5201 Norma… GSM282864     Normal Uninfla… ftp://ftp.ncbi.nl… Reanaly… 109    
# ℹ 192 more rows
# ℹ 53 more variables: current_medication &lt;chr&gt;, birth_date &lt;chr&gt;,
#   ethnicity &lt;chr&gt;, symptoms_onset_date &lt;chr&gt;, diagnosis_date &lt;chr&gt;,
#   joint_problems &lt;chr&gt;, uc_flare_up &lt;chr&gt;, family_history &lt;chr&gt;,
#   aza_treated &lt;chr&gt;, aza_tolerant &lt;chr&gt;, aza_at_present &lt;chr&gt;,
#   procedure_date &lt;chr&gt;, indictation_for_procedure &lt;chr&gt;, ucss &lt;chr&gt;,
#   calprotectin &lt;chr&gt;, esr &lt;chr&gt;, crp &lt;chr&gt;, hb &lt;chr&gt;, wcc &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>The phenodata has been extracted from the gse11223 and now we will pick and choose the data that is, perhaps removing those variables that are of no importance for us, such as status, submission_date, type, channel_count, organism_ch1 (as they are all homo sapiens) and characteristics_ch1</p>
</section>
<section id="selecting-variables" class="level2">
<h2 class="anchored" data-anchor-id="selecting-variables">Selecting variables</h2>
<p>Selecting which variables from phenodata that we would like to keep for further analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata <span class="sc">|&gt;</span> </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geo_accession, patient, current_medication, birth_date, ethnicity, symptoms_onset_date, diagnosis_date, joint_problems, uc_flare_up, family_history, neutrophils, albumin, ibd_affected_relatives, smoking_status, smoking_start_date, smoking_stop_date, smoking_amount, other_illnesses, disease, anatomic_location, inflammation_status, procedure_date)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merging-phenodata" class="level2">
<h2 class="anchored" data-anchor-id="merging-phenodata">Merging phenodata</h2>
<p>Do a left_join() with phenodata and exprs_df on geo_accession</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span> <span class="fu">right_join</span>(phenodata, exprs_filtered, <span class="at">by =</span> <span class="st">"geo_accession"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Stretching the data set so that genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span>  finaldata <span class="sc">|&gt;</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">23</span><span class="sc">:</span><span class="fu">last_col</span>(),</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"gene_id"</span> ,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"gene_expr"</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="featuredatadata-from-_rawgse11223.rds" class="level2">
<h2 class="anchored" data-anchor-id="featuredatadata-from-_rawgse11223.rds">featureData$data from _raw/gse11223.rds</h2>
<p>Metadata concerning genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">pData</span>(<span class="fu">featureData</span>(gse11223))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we filter the same low variance genes from featuredata using the gene ID in genes_keep</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> featuredata <span class="sc">|&gt;</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> genes_keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="exporting-to-data" class="level2">
<h2 class="anchored" data-anchor-id="exporting-to-data">Exporting to data/</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(finaldata, <span class="at">file =</span> <span class="st">"../data/02_dat_clean.csv.gz"</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(featuredata, <span class="at">file =</span> <span class="st">"../data/02_dat_genbank.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-environment" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment">Cleaning environment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="augmenting" class="level1">
<h1>Augmenting</h1>
<section id="loading-libraries-3" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-3">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-finaldata-and-featuredata" class="level2">
<h2 class="anchored" data-anchor-id="loading-finaldata-and-featuredata">Loading finaldata and featuredata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/02_dat_clean.csv.gz"</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/02_dat_genbank.csv.gz"</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(finaldata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "geo_accession"          "patient"                "current_medication"    
 [4] "birth_date"             "ethnicity"              "symptoms_onset_date"   
 [7] "diagnosis_date"         "joint_problems"         "uc_flare_up"           
[10] "family_history"         "neutrophils"            "albumin"               
[13] "ibd_affected_relatives" "smoking_status"         "smoking_start_date"    
[16] "smoking_stop_date"      "smoking_amount"         "other_illnesses"       
[19] "disease"                "anatomic_location"      "inflammation_status"   
[22] "procedure_date"         "gene_id"                "gene_expr"             </code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(featuredata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                   "COL"                  "ROW"                 
 [4] "NAME"                 "SPOT_ID"              "CONTROL_TYPE"        
 [7] "REFSEQ"               "GB_ACC"               "GENE"                
[10] "GENE_SYMBOL"          "GENE_NAME"            "UNIGENE_ID"          
[13] "ENSEMBL_ID"           "TIGR_ID"              "ACCESSION_STRING"    
[16] "CHROMOSOMAL_LOCATION" "CYTOBAND"             "DESCRIPTION"         
[19] "GO_ID"                "SEQUENCE"            </code></pre>
</div>
</div>
</section>
<section id="calculating-the-age-from-date-of-birth" class="level2">
<h2 class="anchored" data-anchor-id="calculating-the-age-from-date-of-birth">Calculating the Age from date of birth</h2>
<p>Birth date reformat - input a date in string format, with “/” as spacer. Date format should be in MM/DD/YY</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="at">file =</span> <span class="st">"99_proj_func.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Making the age variable from the date of birth and the submission date for the article, to estimate the age. at the time of sampling. We identified some birth dates that were written as MM/DD/1060, we belive it is a typo and what was meant was 1960. Hence we changed it to 1960.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">birth_date =</span> <span class="fu">reformat_dates</span>(birth_date)) <span class="sc">|&gt;</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">procedure_date =</span> <span class="fu">as.Date</span>(procedure_date, <span class="st">"%m/%d/%y"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">age =</span> <span class="fu">round</span>(<span class="fu">as.numeric</span>(<span class="fu">difftime</span>(procedure_date, birth_date, <span class="at">units =</span> <span class="st">"days"</span>)) <span class="sc">/</span> <span class="fl">365.25</span>, <span class="dv">0</span>), <span class="at">.after=</span>procedure_date) <span class="sc">|&gt;</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">age_group =</span> <span class="fu">case_when</span>(</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>    <span class="dv">20</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">30</span> <span class="sc">~</span> <span class="st">"(20,30]"</span>,</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>    <span class="dv">30</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">40</span> <span class="sc">~</span> <span class="st">"(30,40]"</span>,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>    <span class="dv">40</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">50</span> <span class="sc">~</span> <span class="st">"(40,50]"</span>,</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>    <span class="dv">50</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">60</span> <span class="sc">~</span> <span class="st">"(50,60]"</span>,</span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>    <span class="dv">60</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">70</span> <span class="sc">~</span> <span class="st">"(60,70]"</span>,</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>    <span class="dv">70</span> <span class="sc">&lt;=</span> age <span class="sc">&amp;</span> age <span class="sc">&lt;</span> <span class="dv">80</span> <span class="sc">~</span> <span class="st">"(70,80]"</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">.after=</span>age)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="reformatting-symptoms_onset_date-diagnosis_date-calculating-the-time-difference-from-symptoms_onset-to-diagnosis_date" class="level2">
<h2 class="anchored" data-anchor-id="reformatting-symptoms_onset_date-diagnosis_date-calculating-the-time-difference-from-symptoms_onset-to-diagnosis_date">Reformatting symptoms_onset_date &amp; diagnosis_date &amp; Calculating the time difference from symptoms_onset to diagnosis_date</h2>
<p>Will reflect time between time of diagnosis and treatment. Shorter time from first symptom to treatment may slow the disease development.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">symptoms_onset_date =</span> <span class="fu">as.Date</span>(symptoms_onset_date, <span class="st">"%m/%d/%y"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diagnosis_date =</span> <span class="fu">as.Date.character</span>(diagnosis_date, <span class="st">"%m/%d/%y"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">time_untreated =</span> <span class="fu">as.numeric</span>(<span class="fu">difftime</span>(diagnosis_date, symptoms_onset_date, <span class="at">units =</span> <span class="st">"days"</span>)), <span class="at">.after =</span> diagnosis_date)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="preparation-for-volcano-plot" class="level2">
<h2 class="anchored" data-anchor-id="preparation-for-volcano-plot">Preparation for volcano plot</h2>
<p>First we remove the NA from both gene_expr and inflammation_status as if we try to do a t-test without removing these NAs it would not work. We are this is only for use in the vulcano plot as it might skew the other analysis. An <code>mutate()</code> <code>inflammation_status</code> in order to create binary factors for the t-test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>final_clean <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_expr),</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span><span class="fu">is.na</span>(inflammation_status)) <span class="sc">|&gt;</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">inflammation_status =</span> <span class="fu">factor</span>(inflammation_status,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Uninflamed"</span>, <span class="st">"Inflamed"</span>)) <span class="co"># control first</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here we get group the data by Gene id. We take the mean of gene expression for both the inflammed group and the uninflammed group. We used tryCatch to make sure the code is run and returns NA if the it fails. We save the significant p-values and negative log 10 p- value.</p>
<p>The t-test significance are done with a 95% confidence interval hence any values outside of the 95 % confidence interval would be considered significant.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> final_clean <span class="sc">|&gt;</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">|&gt;</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># keep only genes with both groups present</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">n_distinct</span>(inflammation_status) <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_inflamed   =</span> <span class="fu">mean</span>(gene_expr[inflammation_status <span class="sc">==</span> <span class="st">"Inflamed"</span>],   <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_uninflamed =</span> <span class="fu">mean</span>(gene_expr[inflammation_status <span class="sc">==</span> <span class="st">"Uninflamed"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># calculate fold change and p values</span></span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">log2FC =</span> mean_inflamed <span class="sc">-</span> mean_uninflamed,</span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">tryCatch</span>(</span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(gene_expr <span class="sc">~</span> inflammation_status)<span class="sc">$</span>p.value,</span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NA_real_</span></span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb57-19"><a href="#cb57-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">neg_log10_p =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value),</span>
<span id="cb57-20"><a href="#cb57-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig =</span> p_value <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb57-21"><a href="#cb57-21" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="preparation-for-random-forest-model" class="level2">
<h2 class="anchored" data-anchor-id="preparation-for-random-forest-model">Preparation for Random Forest model</h2>
<p>Reformatting expression dataframe with classifier (y_value) ready for a Random Forest Classifier model</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>disease_geo <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geo_accession, disease) <span class="sc">|&gt;</span> </span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(geo_accession, disease)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>expr_df <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geo_accession, gene_id, gene_expr) <span class="sc">|&gt;</span> </span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_expr)) <span class="sc">|&gt;</span> </span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id,</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> gene_expr) <span class="sc">|&gt;</span> </span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"Gene_"</span>, .), </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">.cols =</span> <span class="sc">-</span><span class="fu">c</span>(geo_accession)) <span class="sc">|&gt;</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"Gene_"</span>), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>))) <span class="sc">|&gt;</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(geo_accession, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(disease_geo, <span class="at">by =</span> <span class="st">"geo_accession"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_pred_UC =</span> <span class="fu">as.factor</span>(<span class="fu">case_when</span>(disease <span class="sc">==</span> <span class="st">"Normal"</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>                               disease <span class="sc">==</span> <span class="st">"UC"</span> <span class="sc">~</span> <span class="dv">1</span>)),</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">.keep =</span> <span class="st">"unused"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="exporting-to-data-1" class="level2">
<h2 class="anchored" data-anchor-id="exporting-to-data-1">Exporting to data/</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(finaldata, <span class="at">file =</span> <span class="st">"../data/03_dat_aug.csv.gz"</span>)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(featuredata, <span class="at">file =</span> <span class="st">"../data/03_dat_genbank.csv.gz"</span>)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(de_results, <span class="at">file =</span> <span class="st">"../data/03_de_results.csv.gz"</span>)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(expr_df, <span class="at">file =</span> <span class="st">"../data/03_rf_expr_df.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-environment-1" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment-1">Cleaning environment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="describing" class="level1">
<h1>Describing</h1>
<section id="loading-libraries-4" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-4">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"table1"</span>)</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-finaldata-and-featuredata-1" class="level2">
<h2 class="anchored" data-anchor-id="loading-finaldata-and-featuredata-1">Loading finaldata and featuredata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>finaldata   <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_dat_aug.csv.gz"</span>)</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/featuredata.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="finaldata" class="level2">
<h2 class="anchored" data-anchor-id="finaldata">Finaldata:</h2>
<p>Here we select all the the meta data to make a table, so we get an entry for each patient, in order to see their smoking habits, ethnicity, blood status and more.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>fin_all<span class="ot">&lt;-</span>finaldata<span class="sc">|&gt;</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    patient,</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    ethnicity,</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>    joint_problems,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>    uc_flare_up,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>    family_history,</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>    neutrophils,</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>    albumin,</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>    ibd_affected_relatives,</span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>    smoking_status,</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>    smoking_amount,</span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>    other_illnesses,</span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>    disease,</span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>    anatomic_location,</span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>    inflammation_status,</span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>    age,</span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a>    age_group</span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">|&gt;</span></span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(patient,<span class="at">.keep_all=</span><span class="cn">TRUE</span>)<span class="sc">|&gt;</span>  <span class="co"># Keep one record per patient (remove duplicates if multiple biopsies exist)</span></span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">neutrophils=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(neutrophils)),      <span class="co"># Convert to numeric; non-numeric values become NA</span></span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">albumin=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(albumin)),</span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">smoking_amount=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(smoking_amount)),<span class="co"># Smoking amount as numeric</span></span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">age=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(age)),                      <span class="co"># Age converted to numeric</span></span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># fattori</span></span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">ethnicity=</span><span class="fu">factor</span>(ethnicity),                                <span class="co"># Convert categorical variables to factors</span></span>
<span id="cb63-28"><a href="#cb63-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">joint_problems=</span><span class="fu">factor</span>(joint_problems),</span>
<span id="cb63-29"><a href="#cb63-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">uc_flare_up=</span><span class="fu">factor</span>(uc_flare_up),</span>
<span id="cb63-30"><a href="#cb63-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_history=</span><span class="fu">factor</span>(family_history),</span>
<span id="cb63-31"><a href="#cb63-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">ibd_affected_relatives=</span><span class="fu">factor</span>(ibd_affected_relatives),</span>
<span id="cb63-32"><a href="#cb63-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">smoking_status=</span><span class="fu">factor</span>(smoking_status),</span>
<span id="cb63-33"><a href="#cb63-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">disease=</span><span class="fu">factor</span>(disease),</span>
<span id="cb63-34"><a href="#cb63-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">anatomic_location=</span><span class="fu">factor</span>(anatomic_location),</span>
<span id="cb63-35"><a href="#cb63-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">inflammation_status=</span><span class="fu">factor</span>(inflammation_status),</span>
<span id="cb63-36"><a href="#cb63-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">age_group=</span><span class="fu">factor</span>(age_group)</span>
<span id="cb63-37"><a href="#cb63-37" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-38"><a href="#cb63-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-39"><a href="#cb63-39" aria-hidden="true" tabindex="-1"></a>fin_all<span class="sc">$</span>neutrophils[fin_all<span class="sc">$</span>neutrophils<span class="sc">==</span><span class="dv">0</span>]<span class="ot">&lt;-</span><span class="cn">NA</span>   <span class="co"># Replace 0 with NA</span></span>
<span id="cb63-40"><a href="#cb63-40" aria-hidden="true" tabindex="-1"></a>fin_all<span class="sc">$</span>albumin[fin_all<span class="sc">$</span>albumin<span class="sc">==</span><span class="dv">0</span>]<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb63-41"><a href="#cb63-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-42"><a href="#cb63-42" aria-hidden="true" tabindex="-1"></a>fin_all<span class="sc">$</span>other_illnesses_clean<span class="ot">&lt;-</span>dplyr<span class="sc">::</span><span class="fu">case_when</span>(   <span class="co"># Clean comorbidity field into a simple binary variable</span></span>
<span id="cb63-43"><a href="#cb63-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">is.na</span>(fin_all<span class="sc">$</span>other_illnesses)<span class="sc">~</span><span class="cn">NA_character_</span>,</span>
<span id="cb63-44"><a href="#cb63-44" aria-hidden="true" tabindex="-1"></a>  fin_all<span class="sc">$</span>other_illnesses<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"none"</span>,<span class="st">"None"</span>,<span class="st">"NONE"</span>)<span class="sc">~</span><span class="st">"None"</span>,</span>
<span id="cb63-45"><a href="#cb63-45" aria-hidden="true" tabindex="-1"></a>  <span class="cn">TRUE</span><span class="sc">~</span><span class="st">"Any comorbidity"</span></span>
<span id="cb63-46"><a href="#cb63-46" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb63-47"><a href="#cb63-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-48"><a href="#cb63-48" aria-hidden="true" tabindex="-1"></a>fin_all<span class="sc">$</span>other_illnesses_clean<span class="ot">&lt;-</span><span class="fu">factor</span>(             <span class="co"># Convert cleaned comorbidity variable to factor</span></span>
<span id="cb63-49"><a href="#cb63-49" aria-hidden="true" tabindex="-1"></a>  fin_all<span class="sc">$</span>other_illnesses_clean,</span>
<span id="cb63-50"><a href="#cb63-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"None"</span>,<span class="st">"Any comorbidity"</span>)</span>
<span id="cb63-51"><a href="#cb63-51" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb63-52"><a href="#cb63-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-53"><a href="#cb63-53" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>neutrophils)<span class="ot">&lt;-</span><span class="st">"Neutrophils"</span>                  <span class="co"># Labels for table1() formatting</span></span>
<span id="cb63-54"><a href="#cb63-54" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>albumin)<span class="ot">&lt;-</span><span class="st">"Albumin"</span></span>
<span id="cb63-55"><a href="#cb63-55" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>ethnicity)<span class="ot">&lt;-</span><span class="st">"Ethnicity"</span></span>
<span id="cb63-56"><a href="#cb63-56" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>joint_problems)<span class="ot">&lt;-</span><span class="st">"Joint problems"</span></span>
<span id="cb63-57"><a href="#cb63-57" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>uc_flare_up)<span class="ot">&lt;-</span><span class="st">"UC flare-up"</span></span>
<span id="cb63-58"><a href="#cb63-58" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>family_history)<span class="ot">&lt;-</span><span class="st">"Family history of IBD"</span></span>
<span id="cb63-59"><a href="#cb63-59" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>ibd_affected_relatives)<span class="ot">&lt;-</span><span class="st">"IBD-affected relatives"</span></span>
<span id="cb63-60"><a href="#cb63-60" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>smoking_status)<span class="ot">&lt;-</span><span class="st">"Smoking status"</span></span>
<span id="cb63-61"><a href="#cb63-61" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>other_illnesses_clean)<span class="ot">&lt;-</span><span class="st">"Other illnesses"</span></span>
<span id="cb63-62"><a href="#cb63-62" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>disease)<span class="ot">&lt;-</span><span class="st">"Disease group"</span></span>
<span id="cb63-63"><a href="#cb63-63" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>anatomic_location)<span class="ot">&lt;-</span><span class="st">"Anatomic location"</span></span>
<span id="cb63-64"><a href="#cb63-64" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>inflammation_status)<span class="ot">&lt;-</span><span class="st">"Inflammation status"</span></span>
<span id="cb63-65"><a href="#cb63-65" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>age)<span class="ot">&lt;-</span><span class="st">"Age"</span></span>
<span id="cb63-66"><a href="#cb63-66" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(fin_all<span class="sc">$</span>age_group)<span class="ot">&lt;-</span><span class="st">"Age group"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here we create a table of the data using the <code>table1</code> package using the variables shown in the code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table1</span>(</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> age <span class="sc">+</span> neutrophils <span class="sc">+</span> albumin <span class="sc">+</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>    ethnicity <span class="sc">+</span> smoking_status <span class="sc">+</span> other_illnesses_clean <span class="sc">+</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>    anatomic_location <span class="sc">+</span> inflammation_status <span class="sc">+</span> joint_problems <span class="sc">+</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>    family_history <span class="sc">+</span> ibd_affected_relatives <span class="sc">+</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>    age_group</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span> disease,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fin_all</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="Rtable1">
<table class="Rtable1 caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th class="rowlabel firstrow lastrow" data-quarto-table-cell-role="th"></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">Normal<br>
<span class="stratn">(N=30)</span></span></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">UC<br>
<span class="stratn">(N=63)</span></span></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">Overall<br>
<span class="stratn">(N=93)</span></span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="rowlabel firstrow">Age</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">Mean (SD)</td>
<td>41.3 (16.3)</td>
<td>50.9 (15.9)</td>
<td>47.8 (16.6)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Median [Min, Max]</td>
<td class="lastrow">41.5 [17.0, 76.0]</td>
<td class="lastrow">50.0 [19.0, 84.0]</td>
<td class="lastrow">47.0 [17.0, 84.0]</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Neutrophils</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">Mean (SD)</td>
<td>NA (NA)</td>
<td>7.00 (6.06)</td>
<td>7.00 (6.06)</td>
</tr>
<tr class="even">
<td class="rowlabel">Median [Min, Max]</td>
<td>NA [NA, NA]</td>
<td>4.50 [3.00, 16.0]</td>
<td>4.50 [3.00, 16.0]</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">30 (100%)</td>
<td class="lastrow">59 (93.7%)</td>
<td class="lastrow">89 (95.7%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Albumin</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">Mean (SD)</td>
<td>NA (NA)</td>
<td>35.0 (10.6)</td>
<td>35.0 (10.6)</td>
</tr>
<tr class="even">
<td class="rowlabel">Median [Min, Max]</td>
<td>NA [NA, NA]</td>
<td>39.0 [23.0, 43.0]</td>
<td>39.0 [23.0, 43.0]</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">30 (100%)</td>
<td class="lastrow">60 (95.2%)</td>
<td class="lastrow">90 (96.8%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Ethnicity</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">ASIAN</td>
<td>0 (0%)</td>
<td>3 (4.8%)</td>
<td>3 (3.2%)</td>
</tr>
<tr class="even">
<td class="rowlabel">CAUCASIAN</td>
<td>24 (80.0%)</td>
<td>60 (95.2%)</td>
<td>84 (90.3%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">JEWISH</td>
<td class="lastrow">6 (20.0%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">6 (6.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Smoking status</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">current</td>
<td>0 (0%)</td>
<td>6 (9.5%)</td>
<td>6 (6.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel">ex</td>
<td>0 (0%)</td>
<td>22 (34.9%)</td>
<td>22 (23.7%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">never</td>
<td>0 (0%)</td>
<td>33 (52.4%)</td>
<td>33 (35.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel">unknown</td>
<td>0 (0%)</td>
<td>2 (3.2%)</td>
<td>2 (2.2%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">30 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">30 (32.3%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Other illnesses</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">None</td>
<td>0 (0%)</td>
<td>38 (60.3%)</td>
<td>38 (40.9%)</td>
</tr>
<tr class="even">
<td class="rowlabel">Any comorbidity</td>
<td>0 (0%)</td>
<td>25 (39.7%)</td>
<td>25 (26.9%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">30 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">30 (32.3%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Anatomic location</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">ascending colon</td>
<td>4 (13.3%)</td>
<td>12 (19.0%)</td>
<td>16 (17.2%)</td>
</tr>
<tr class="even">
<td class="rowlabel">descending colon</td>
<td>3 (10.0%)</td>
<td>10 (15.9%)</td>
<td>13 (14.0%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">sigmoid colon</td>
<td>20 (66.7%)</td>
<td>38 (60.3%)</td>
<td>58 (62.4%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">terminal ileum</td>
<td class="lastrow">3 (10.0%)</td>
<td class="lastrow">3 (4.8%)</td>
<td class="lastrow">6 (6.5%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Inflammation status</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">Inflamed</td>
<td>3 (10.0%)</td>
<td>34 (54.0%)</td>
<td>37 (39.8%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Uninflamed</td>
<td class="lastrow">27 (90.0%)</td>
<td class="lastrow">29 (46.0%)</td>
<td class="lastrow">56 (60.2%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Joint problems</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">FALSE</td>
<td>30 (100%)</td>
<td>62 (98.4%)</td>
<td>92 (98.9%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">TRUE</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">1 (1.6%)</td>
<td class="lastrow">1 (1.1%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Family history of IBD</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">FALSE</td>
<td>30 (100%)</td>
<td>57 (90.5%)</td>
<td>87 (93.5%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">TRUE</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">6 (9.5%)</td>
<td class="lastrow">6 (6.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">IBD-affected relatives</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">0</td>
<td>0 (0%)</td>
<td>59 (93.7%)</td>
<td>59 (63.4%)</td>
</tr>
<tr class="even">
<td class="rowlabel">1</td>
<td>0 (0%)</td>
<td>2 (3.2%)</td>
<td>2 (2.2%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">2</td>
<td>0 (0%)</td>
<td>2 (3.2%)</td>
<td>2 (2.2%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">30 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">30 (32.3%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Age group</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">(20,30]</td>
<td>7 (23.3%)</td>
<td>7 (11.1%)</td>
<td>14 (15.1%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">(30,40]</td>
<td>4 (13.3%)</td>
<td>4 (6.3%)</td>
<td>8 (8.6%)</td>
</tr>
<tr class="even">
<td class="rowlabel">(40,50]</td>
<td>9 (30.0%)</td>
<td>18 (28.6%)</td>
<td>27 (29.0%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">(50,60]</td>
<td>4 (13.3%)</td>
<td>11 (17.5%)</td>
<td>15 (16.1%)</td>
</tr>
<tr class="even">
<td class="rowlabel">(60,70]</td>
<td>2 (6.7%)</td>
<td>14 (22.2%)</td>
<td>16 (17.2%)</td>
</tr>
<tr class="odd">
<td class="rowlabel">(70,80]</td>
<td>2 (6.7%)</td>
<td>4 (6.3%)</td>
<td>6 (6.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">2 (6.7%)</td>
<td class="lastrow">5 (7.9%)</td>
<td class="lastrow">7 (7.5%)</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="finaldata-interpretation" class="level2">
<h2 class="anchored" data-anchor-id="finaldata-interpretation">Finaldata interpretation:</h2>
<p>Ninety-three subjects were included, of whom 63 (67.7%) were diagnosed with ulcerative colitis (UC) and 30 (32.3%) with histologically normal mucosa (controls).</p>
<p>The mean age of the UC group was 50.9 ± 15.9 years, while the control group had a median age of 47 years (range 17-84). In terms of age distribution, the trend is for a higher percentage of subjects older than 60 years in the UC group (22.2% vs.&nbsp;6.7%).</p>
<p>Laboratory parameters for neutrophilis and albumin were present only in a small portion of the UC group, in 4 and 3 subjects respectively, and were completely absent in the control group. They will therefore not be considered in subsequent investigations.</p>
<p>The predominant ethnicity in the dataset is Caucasian (90.3%), followed by Asian (3.2% all in the UC group) and Jewish (6.5% all in the control group)</p>
<p>In the UC group, 9.5% were current smokers, 34.9% were former smokers, and 52.4% had never smoked (and 3.2% were unknown). While in the control group, this variable was absent in 100% of cases.</p>
<p>Similarly, other concomitant conditions were reported only in the UC group, where 60.3% had no comorbidities and 39.7% reported at least one extraintestinal condition, while this information was missing in 100% of controls.</p>
<p>The most frequently sampled anatomical sites are approximately equal in the two groups and include:</p>
<ul>
<li><p>sigmoid colon (62.4%)</p></li>
<li><p>ascending colon (17.2%)</p></li>
<li><p>descending colon (14.0%)</p></li>
<li><p>terminal ileum (6.5%)</p></li>
</ul>
<p>In the UC group, 54.0% of the samples came from macroscopically inflamed mucosa, while in the control group only 10%</p>
<p>Almost all subjects in the dataset (98.8% and 93.5%, respectively) did not have any joint problems or a family history of inflammatory bowel disease (IBD).</p>
<p>As for relatives affected by IBD, they are present only in the UC group (4 patients with 1 or 2 affected relatives), while they are totally absent in the control group.</p>
</section>
<section id="featuredata" class="level2">
<h2 class="anchored" data-anchor-id="featuredata">Featuredata:</h2>
<p>here we gather featuredata and investigate which information is available about the gene meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>feat_all<span class="ot">&lt;-</span>featuredata<span class="sc">|&gt;</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">ID=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(ID)), <span class="co"># Convert ID fields to numeric</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">COL=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(COL)),  <span class="co"># Array column index</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">ROW=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(ROW)), <span class="co"># Array row index</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">GENE=</span><span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(GENE)), <span class="co"># Internal gene ID</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">CONTROL_TYPE=</span><span class="fu">factor</span>(CONTROL_TYPE), <span class="co"># Convert control-type field to factor</span></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create binary indicators (Yes/No) for the presence of annotation fields</span></span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_NAME=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(NAME)<span class="sc">|</span>NAME<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_SPOT_ID=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(SPOT_ID)<span class="sc">|</span>SPOT_ID<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_REFSEQ=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(REFSEQ)<span class="sc">|</span>REFSEQ<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_GB_ACC=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(GB_ACC)<span class="sc">|</span>GB_ACC<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_GENE_SYMBOL=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(GENE_SYMBOL)<span class="sc">|</span>GENE_SYMBOL<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_GENE_NAME=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(GENE_NAME)<span class="sc">|</span>GENE_NAME<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_UNIGENE=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(UNIGENE_ID)<span class="sc">|</span>UNIGENE_ID<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-18"><a href="#cb65-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_ENSEMBL=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(ENSEMBL_ID)<span class="sc">|</span>ENSEMBL_ID<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-19"><a href="#cb65-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_TIGR=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(TIGR_ID)<span class="sc">|</span>TIGR_ID<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-20"><a href="#cb65-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_ACCESSION=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(ACCESSION_STRING)<span class="sc">|</span>ACCESSION_STRING<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-21"><a href="#cb65-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_CHR_LOC=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(CHROMOSOMAL_LOCATION)<span class="sc">|</span>CHROMOSOMAL_LOCATION<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-22"><a href="#cb65-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_CYTOBAND=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(CYTOBAND)<span class="sc">|</span>CYTOBAND<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-23"><a href="#cb65-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_DESCRIPTION=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(DESCRIPTION)<span class="sc">|</span>DESCRIPTION<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-24"><a href="#cb65-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_GO=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(GO_ID)<span class="sc">|</span>GO_ID<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>)),</span>
<span id="cb65-25"><a href="#cb65-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">HAS_SEQUENCE=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">is.na</span>(SEQUENCE)<span class="sc">|</span>SEQUENCE<span class="sc">==</span><span class="st">""</span>,<span class="st">"No"</span>,<span class="st">"Yes"</span>))</span>
<span id="cb65-26"><a href="#cb65-26" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb65-27"><a href="#cb65-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-28"><a href="#cb65-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Labels for table1</span></span>
<span id="cb65-29"><a href="#cb65-29" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>ID)<span class="ot">&lt;-</span><span class="st">"Probe ID"</span></span>
<span id="cb65-30"><a href="#cb65-30" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>COL)<span class="ot">&lt;-</span><span class="st">"Array column index"</span></span>
<span id="cb65-31"><a href="#cb65-31" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>ROW)<span class="ot">&lt;-</span><span class="st">"Array row index"</span></span>
<span id="cb65-32"><a href="#cb65-32" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>GENE)<span class="ot">&lt;-</span><span class="st">"Internal gene ID"</span></span>
<span id="cb65-33"><a href="#cb65-33" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>CONTROL_TYPE)<span class="ot">&lt;-</span><span class="st">"Control type"</span></span>
<span id="cb65-34"><a href="#cb65-34" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_NAME)<span class="ot">&lt;-</span><span class="st">"Has probe NAME"</span></span>
<span id="cb65-35"><a href="#cb65-35" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_SPOT_ID)<span class="ot">&lt;-</span><span class="st">"Has SPOT_ID"</span></span>
<span id="cb65-36"><a href="#cb65-36" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_REFSEQ)<span class="ot">&lt;-</span><span class="st">"Has RefSeq ID"</span></span>
<span id="cb65-37"><a href="#cb65-37" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_GB_ACC)<span class="ot">&lt;-</span><span class="st">"Has GenBank accession"</span></span>
<span id="cb65-38"><a href="#cb65-38" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_GENE_SYMBOL)<span class="ot">&lt;-</span><span class="st">"Has gene symbol"</span></span>
<span id="cb65-39"><a href="#cb65-39" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_GENE_NAME)<span class="ot">&lt;-</span><span class="st">"Has gene name"</span></span>
<span id="cb65-40"><a href="#cb65-40" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_UNIGENE)<span class="ot">&lt;-</span><span class="st">"Has UniGene ID"</span></span>
<span id="cb65-41"><a href="#cb65-41" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_ENSEMBL)<span class="ot">&lt;-</span><span class="st">"Has Ensembl ID"</span></span>
<span id="cb65-42"><a href="#cb65-42" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_TIGR)<span class="ot">&lt;-</span><span class="st">"Has TIGR ID"</span></span>
<span id="cb65-43"><a href="#cb65-43" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_ACCESSION)<span class="ot">&lt;-</span><span class="st">"Has accession string"</span></span>
<span id="cb65-44"><a href="#cb65-44" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_CHR_LOC)<span class="ot">&lt;-</span><span class="st">"Has chromosomal location"</span></span>
<span id="cb65-45"><a href="#cb65-45" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_CYTOBAND)<span class="ot">&lt;-</span><span class="st">"Has cytoband"</span></span>
<span id="cb65-46"><a href="#cb65-46" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_DESCRIPTION)<span class="ot">&lt;-</span><span class="st">"Has description"</span></span>
<span id="cb65-47"><a href="#cb65-47" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_GO)<span class="ot">&lt;-</span><span class="st">"Has GO annotation"</span></span>
<span id="cb65-48"><a href="#cb65-48" aria-hidden="true" tabindex="-1"></a><span class="fu">label</span>(feat_all<span class="sc">$</span>HAS_SEQUENCE)<span class="ot">&lt;-</span><span class="st">"Has probe sequence"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Creating a table for summarizing the gene meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table1</span>(</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> CONTROL_TYPE <span class="sc">+</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>    HAS_REFSEQ <span class="sc">+</span> HAS_GB_ACC <span class="sc">+</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>    HAS_GENE_NAME <span class="sc">+</span> HAS_UNIGENE <span class="sc">+</span> HAS_ENSEMBL <span class="sc">+</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>    HAS_TIGR <span class="sc">+</span> HAS_ACCESSION <span class="sc">+</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>    HAS_CHR_LOC <span class="sc">+</span> HAS_CYTOBAND <span class="sc">+</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>    HAS_DESCRIPTION <span class="sc">+</span> HAS_GO <span class="sc">+</span> HAS_SEQUENCE</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span> HAS_GENE_SYMBOL,</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> feat_all</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="Rtable1">
<table class="Rtable1 caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th class="rowlabel firstrow lastrow" data-quarto-table-cell-role="th"></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">No<br>
<span class="stratn">(N=881)</span></span></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">Yes<br>
<span class="stratn">(N=2028)</span></span></th>
<th class="firstrow lastrow" data-quarto-table-cell-role="th"><span class="stratlabel">Overall<br>
<span class="stratn">(N=2909)</span></span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="rowlabel firstrow">Control type</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">FALSE</td>
<td class="lastrow">881 (100%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2909 (100%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has RefSeq ID</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>814 (92.4%)</td>
<td>167 (8.2%)</td>
<td>981 (33.7%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">67 (7.6%)</td>
<td class="lastrow">1861 (91.8%)</td>
<td class="lastrow">1928 (66.3%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has GenBank accession</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">No</td>
<td>421 (47.8%)</td>
<td>0 (0%)</td>
<td>421 (14.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">460 (52.2%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2488 (85.5%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has gene name</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>881 (100%)</td>
<td>0 (0%)</td>
<td>881 (30.3%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2028 (69.7%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has UniGene ID</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">No</td>
<td>472 (53.6%)</td>
<td>23 (1.1%)</td>
<td>495 (17.0%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">409 (46.4%)</td>
<td class="lastrow">2005 (98.9%)</td>
<td class="lastrow">2414 (83.0%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has Ensembl ID</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>733 (83.2%)</td>
<td>275 (13.6%)</td>
<td>1008 (34.7%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">148 (16.8%)</td>
<td class="lastrow">1753 (86.4%)</td>
<td class="lastrow">1901 (65.3%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has TIGR ID</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">No</td>
<td>402 (45.6%)</td>
<td>77 (3.8%)</td>
<td>479 (16.5%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">479 (54.4%)</td>
<td class="lastrow">1951 (96.2%)</td>
<td class="lastrow">2430 (83.5%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has accession string</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>240 (27.2%)</td>
<td>0 (0%)</td>
<td>240 (8.3%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">641 (72.8%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2669 (91.7%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has chromosomal location</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">No</td>
<td>42 (4.8%)</td>
<td>6 (0.3%)</td>
<td>48 (1.7%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">839 (95.2%)</td>
<td class="lastrow">2022 (99.7%)</td>
<td class="lastrow">2861 (98.4%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has cytoband</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>61 (6.9%)</td>
<td>15 (0.7%)</td>
<td>76 (2.6%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">820 (93.1%)</td>
<td class="lastrow">2013 (99.3%)</td>
<td class="lastrow">2833 (97.4%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has description</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel">No</td>
<td>342 (38.8%)</td>
<td>0 (0%)</td>
<td>342 (11.8%)</td>
</tr>
<tr class="even">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">539 (61.2%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2567 (88.2%)</td>
</tr>
<tr class="odd">
<td class="rowlabel firstrow">Has GO annotation</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="even">
<td class="rowlabel">No</td>
<td>821 (93.2%)</td>
<td>427 (21.1%)</td>
<td>1248 (42.9%)</td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">60 (6.8%)</td>
<td class="lastrow">1601 (78.9%)</td>
<td class="lastrow">1661 (57.1%)</td>
</tr>
<tr class="even">
<td class="rowlabel firstrow">Has probe sequence</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr class="odd">
<td class="rowlabel lastrow">Yes</td>
<td class="lastrow">881 (100%)</td>
<td class="lastrow">2028 (100%)</td>
<td class="lastrow">2909 (100%)</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="featuredata-interpretation" class="level2">
<h2 class="anchored" data-anchor-id="featuredata-interpretation">Featuredata interpretation:</h2>
<p>This table does not tell us anything about expression values ​​or UC vs control differences, however it can provide us with valuable information regarding the quality of the probe annotations.</p>
<ul>
<li><p>Group sizes:<br>
Normal group: 881<br>
UC group: 2028<br>
Overall: 2909</p></li>
<li><p>Control type (CONTROL_TYPE = FALSE):<br>
Normal group: 881/881 (100%)<br>
UC group: 2028/2028 (100%)<br>
Overall: 2909/2909 (100%)</p></li>
<li><p>Has RefSeq ID:<br>
Normal group: No 814 (92.4%), Yes 67 (7.6%)<br>
UC group: No 167 (8.2%), Yes 1861 (91.8%)<br>
Overall: No 981 (33.7%), Yes 1928 (66.3%)</p></li>
<li><p>Has GenBank accession:<br>
Normal group: No 421 (47.8%), Yes 460 (52.2%)<br>
UC group: No 0 (0%), Yes 2028 (100%)<br>
Overall: No 421 (14.5%), Yes 2488 (85.5%)</p></li>
<li><p>Has gene name:<br>
Normal group: No 881 (100%), Yes 0 (0%)<br>
UC group: No 0 (0%), Yes 2028 (100%)<br>
Overall: No 881 (30.3%), Yes 2028 (69.7%)</p></li>
<li><p>Has UniGene ID:<br>
Normal group: No 472 (53.6%), Yes 409 (46.4%)<br>
UC group: No 23 (1.1%), Yes 2005 (98.9%)<br>
Overall: No 495 (17.0%), Yes 2414 (83.0%)</p></li>
<li><p>Has Ensembl ID:<br>
Normal group: No 733 (83.2%), Yes 148 (16.8%)<br>
UC group: No 275 (13.6%), Yes 1753 (86.4%)<br>
Overall: No 1008 (34.7%), Yes 1901 (65.3%)</p></li>
<li><p>Has TIGR ID:<br>
Normal group: No 402 (45.6%), Yes 479 (54.4%)<br>
UC group: No 77 (3.8%), Yes 1951 (96.2%)<br>
Overall: No 479 (16.5%), Yes 2430 (83.5%)</p></li>
<li><p>Has accession string:<br>
Normal group: No 240 (27.2%), Yes 641 (72.8%)<br>
UC group: No 0 (0%), Yes 2028 (100%)<br>
Overall: No 240 (8.3%), Yes 2669 (91.7%)</p></li>
<li><p>Has chromosomal location:<br>
Normal group: No 42 (4.8%), Yes 839 (95.2%)<br>
UC group: No 6 (0.3%), Yes 2022 (99.7%)<br>
Overall: No 48 (1.7%), Yes 2861 (98.4%)</p></li>
<li><p>Has cytoband:<br>
Normal group: No 61 (6.9%), Yes 820 (93.1%)<br>
UC group: No 15 (0.7%), Yes 2013 (99.3%)<br>
Overall: No 76 (2.6%), Yes 2833 (97.4%)</p></li>
<li><p>Has description:<br>
Normal group: No 342 (38.8%), Yes 539 (61.2%)<br>
UC group: No 0 (0%), Yes 2028 (100%)<br>
Overall: No 342 (11.8%), Yes 2567 (88.2%)</p></li>
<li><p>Has GO annotation:<br>
Normal group: No 821 (93.2%), Yes 60 (6.8%)<br>
UC group: No 427 (21.1%), Yes 1601 (78.9%)<br>
Overall: No 1248 (42.9%), Yes 1661 (57.1%)</p></li>
<li><p>Has probe sequence:<br>
Normal group: Yes 881 (100%)<br>
UC group: Yes 2028 (100%)<br>
Overall: Yes 2909 (100%)</p></li>
</ul>
</section>
<section id="cleaning-environment-2" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment-2">Cleaning environment:</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="analysis-of-gene-expression" class="level1">
<h1>Analysis of gene expression</h1>
<section id="loading-libraries-5" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-5">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-finaldata-and-featuredata-2" class="level2">
<h2 class="anchored" data-anchor-id="loading-finaldata-and-featuredata-2">Loading finaldata and featuredata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_dat_aug.csv.gz"</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/featuredata.csv.gz"</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"99_proj_func.R"</span>)</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(finaldata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "geo_accession"          "patient"                "current_medication"    
 [4] "birth_date"             "ethnicity"              "symptoms_onset_date"   
 [7] "diagnosis_date"         "time_untreated"         "joint_problems"        
[10] "uc_flare_up"            "family_history"         "neutrophils"           
[13] "albumin"                "ibd_affected_relatives" "smoking_status"        
[16] "smoking_start_date"     "smoking_stop_date"      "smoking_amount"        
[19] "other_illnesses"        "disease"                "anatomic_location"     
[22] "inflammation_status"    "procedure_date"         "age"                   
[25] "age_group"              "gene_id"                "gene_expr"             </code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(featuredata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                   "COL"                  "ROW"                 
 [4] "NAME"                 "SPOT_ID"              "CONTROL_TYPE"        
 [7] "REFSEQ"               "GB_ACC"               "GENE"                
[10] "GENE_SYMBOL"          "GENE_NAME"            "UNIGENE_ID"          
[13] "ENSEMBL_ID"           "TIGR_ID"              "ACCESSION_STRING"    
[16] "CHROMOSOMAL_LOCATION" "CYTOBAND"             "DESCRIPTION"         
[19] "GO_ID"                "SEQUENCE"            </code></pre>
</div>
</div>
</section>
<section id="looking-at-age-groups" class="level2">
<h2 class="anchored" data-anchor-id="looking-at-age-groups">Looking at age groups</h2>
<section id="correlation-between-age-group-and-diagnosis-time" class="level3">
<h3 class="anchored" data-anchor-id="correlation-between-age-group-and-diagnosis-time">Correlation between age group and diagnosis time</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(geo_accession, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age_group)) <span class="sc">|&gt;</span></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> age_group, <span class="at">fill =</span> inflammation_status))<span class="sc">+</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="fu">position_dodge2</span>(<span class="at">preserve =</span> <span class="st">"single"</span>, <span class="at">padding =</span> <span class="dv">0</span>), </span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.4</span>)<span class="sc">+</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>     ) <span class="sc">+</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">labs</span>(</span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Age groups"</span>,</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"n"</span>,</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Inflammation-status"</span></span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>      )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/age_groups_distribution_inflammation_status-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"age_groups_distribution_inflammation_status.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="age-group-and-diagnosis-time" class="level2">
<h2 class="anchored" data-anchor-id="age-group-and-diagnosis-time">Age group and diagnosis time</h2>
<p>Plotting the correlation between the different age groups and diagnosis time to see, if there is a correlation between the two. Although the medians in the age groups are all similar, some patients in the age group (50,60) takes a long time to get diagnosed. The patients in the age group (30,40] takes the shortest time to get diagnosed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span>  </span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age_group)) <span class="sc">|&gt;</span> </span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_untreated)) <span class="sc">|&gt;</span> </span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time_untreated <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(age_group) <span class="sc">|&gt;</span> </span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 2
  age_group     n
  &lt;chr&gt;     &lt;int&gt;
1 (20,30]   31999
2 (30,40]   17454
3 (40,50]   98906
4 (50,60]   43635
5 (60,70]   46544
6 (70,80]    8727</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span>  </span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age_group)) <span class="sc">|&gt;</span> </span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_untreated)) <span class="sc">|&gt;</span> </span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time_untreated <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(age_group) <span class="sc">|&gt;</span> </span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> age_group, <span class="at">y =</span> time_untreated)) <span class="sc">+</span></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Age Groups"</span>,</span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Time to diagnostics (days)"</span>,</span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Time from 1st UC symtomps to diagnostics against age"</span>) <span class="sc">+</span></span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/time_to_diagnosis_vs_age-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"time_to_diagnosis_vs_age.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="impact-of-smoking-on-inflammation-status" class="level2">
<h2 class="anchored" data-anchor-id="impact-of-smoking-on-inflammation-status">Impact of smoking on inflammation status</h2>
<section id="correlation-between-gene-expression-inflammation-status-and-smoking-status" class="level3">
<h3 class="anchored" data-anchor-id="correlation-between-gene-expression-inflammation-status-and-smoking-status">Correlation between gene expression, inflammation status and smoking status</h3>
<p>Plotting the correlation between gene expression, inflammation status and smoking status. Most patients are either ex-smokers or have never smoked. The group of patients thats do smoke as well as the ones that have never smoked have a slightly higher subset that are inflamed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(geo_accession, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(smoking_status)) <span class="sc">|&gt;</span> </span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> smoking_status, </span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> inflammation_status)) <span class="sc">+</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="fu">position_dodge2</span>(<span class="at">preserve =</span> <span class="st">"single"</span>, <span class="at">padding =</span> <span class="dv">0</span>),</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>     ) <span class="sc">+</span></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Smoking status"</span>, <span class="at">y =</span> <span class="st">"n"</span>, <span class="at">fill =</span> <span class="st">"Inflammation-status"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/gene_expression_by_smoking_status_and_inflammation-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"gene_expression_by_smoking_status_and_inflammation.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="correlation-between-smoking-and-the-diagnosis-time" class="level3">
<h3 class="anchored" data-anchor-id="correlation-between-smoking-and-the-diagnosis-time">Correlation between smoking and the diagnosis time</h3>
<p>Plotting to see the correlation between patients that are smokers and how long it takes for them to get diagnosed. The patient group that whose smoking habits are unknown takes the shortest time to get diagnosed, followed by the current smokers. The group of patients that have never smoked takes the longest time to get diagnosed. However, since the “smoker” group and the “unknown” group are the much smaller than the “ex” and the “never” group, the results may be unreliable. Moreover, there are two three outliers in the plot who have all taken a very long time to get diagnosed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span>  </span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(smoking_status)) <span class="sc">|&gt;</span> </span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_untreated)) <span class="sc">|&gt;</span> </span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time_untreated <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(smoking_status) <span class="sc">|&gt;</span> </span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 2
  smoking_status      n
  &lt;chr&gt;           &lt;int&gt;
1 current         17454
2 ex             113451
3 never          116360
4 unknown          8727</code></pre>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span>  </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(smoking_status)) <span class="sc">|&gt;</span> </span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_untreated)) <span class="sc">|&gt;</span> </span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time_untreated <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(smoking_status) <span class="sc">|&gt;</span> </span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> smoking_status, <span class="at">y =</span> time_untreated)) <span class="sc">+</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Smoking Status"</span>,</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Time to diagnostics (days)"</span>,</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Time from 1st UC symtomps to diagnostics against smoking"</span>) <span class="sc">+</span></span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/time_to_diagnosis_vs_smoking_status-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"time_to_diagnosis_vs_smoking_status.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="significance-based-on-anatomical-location" class="level2">
<h2 class="anchored" data-anchor-id="significance-based-on-anatomical-location">Significance based on anatomical location</h2>
<p>The graph shows that significant variations in expression between inflamed and non-inflamed samples appear only in the descending colon and sigmoid colon. In the ascending colon, the differences do not exceed the significance threshold, while in the terminal ileum, the inflamed samples are too few to make a reliable comparison.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>de_by_loc <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">complete.cases</span>(gene_expr, inflammation_status, anatomic_location)) <span class="sc">|&gt;</span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">inflammation_status =</span> <span class="fu">factor</span>(inflammation_status,</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">c</span>(<span class="st">"Uninflamed"</span>, <span class="st">"Inflamed"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(anatomic_location, gene_id) <span class="sc">|&gt;</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Keep only genes where both groups are present in that location</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">n_distinct</span>(inflammation_status) <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">log2FC  =</span> <span class="fu">mean</span>(gene_expr[inflammation_status <span class="sc">==</span> <span class="st">"Inflamed"</span>],   <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">-</span></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mean</span>(gene_expr[inflammation_status <span class="sc">==</span> <span class="st">"Uninflamed"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">tryCatch</span>(</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(gene_expr <span class="sc">~</span> inflammation_status)<span class="sc">$</span>p.value,</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NA_real_</span></span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(anatomic_location) <span class="sc">|&gt;</span></span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">p.adjust</span>(p_value, <span class="st">"BH"</span>),</span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig =</span> <span class="sc">!</span><span class="fu">is.na</span>(FDR) <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>de_by_loc <span class="sc">|&gt;</span></span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(anatomic_location, sig) <span class="sc">|&gt;</span></span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">|&gt;</span></span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(anatomic_location, n, <span class="at">fill =</span> sig)) <span class="sc">+</span></span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb85-31"><a href="#cb85-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/significant_gene_counts_by_anatomic_location-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"significant_gene_count_by_anatomic_location.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-environment-3" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment-3">Cleaning environment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="analysis-of-metadata" class="level1">
<h1>Analysis of metadata</h1>
<section id="loading-libraries-6" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-6">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-finaldata-featuredata-and-de_results" class="level2">
<h2 class="anchored" data-anchor-id="loading-finaldata-featuredata-and-de_results">Loading finaldata, featuredata and de_results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>finaldata   <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_dat_aug.csv.gz"</span>)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/featuredata.csv.gz"</span>)</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_de_results.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="overall-gene-expression-distribution" class="level2">
<h2 class="anchored" data-anchor-id="overall-gene-expression-distribution">Overall gene expression distribution</h2>
<p>To get an idea of the distribution of the genes between samples categorized as either inflamed or uninflamed, we make a histogram (see Figure X), which shows that they have a normal distribution around with a mean slightly above 0. However both distributions appear similar, though the count of genes with uninflamed is higher.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">inflammation_status =</span> <span class="fu">factor</span>(inflammation_status,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Uninflamed"</span>, <span class="st">"Inflamed"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> gene_expr, <span class="at">fill =</span> inflammation_status)) <span class="sc">+</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">bins =</span> <span class="dv">80</span>) <span class="sc">+</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Inflammation state"</span>,</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Distribution of gene expression in inflamed and uninflamed samples"</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47900 rows containing non-finite outside the scale range
(`stat_bin()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/distribution_gene_expression_inflamed_uninflamed-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"distribution_gene_expression_inflamed_uninflamed.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47900 rows containing non-finite outside the scale range
(`stat_bin()`).</code></pre>
</div>
</div>
</section>
<section id="expression-by-anatomic-location-and-inflammation" class="level2">
<h2 class="anchored" data-anchor-id="expression-by-anatomic-location-and-inflammation"><strong>Expression by anatomic location and inflammation</strong></h2>
<p>We then try to get an idea, if there is a difference in the gene expression pattern across anatomic locations i.e.&nbsp;different parts of the intestinal tract and whether or not the sample was inflamed or not. From this plot it is not clear, so we explore further.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>loc_counts <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(anatomic_location)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> anatomic_location, <span class="at">y =</span> gene_expr, </span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> inflammation_status)) <span class="sc">+</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="cf">function</span>(x) {</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(x, <span class="st">" (n = "</span>, loc_counts<span class="sc">$</span>n[<span class="fu">match</span>(x, loc_counts<span class="sc">$</span>anatomic_location)], <span class="st">")"</span>)</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Anatomic location"</span>,</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Gene expression"</span>,</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Inflammation state"</span></span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47900 rows containing non-finite outside the scale range
(`stat_ydensity()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47900 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/gene_expression_by_anatomic_location_and_inflammation-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"gene_expression_by_anatomic_location_and_inflammation.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47900 rows containing non-finite outside the scale range
(`stat_ydensity()`).
Removed 47900 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
</div>
</section>
<section id="mean-gene-expression-per-patient-across-in-inflamed-and-un-inflamed-samples" class="level2">
<h2 class="anchored" data-anchor-id="mean-gene-expression-per-patient-across-in-inflamed-and-un-inflamed-samples">Mean gene expression per patient across in inflamed and un-inflamed samples</h2>
<p>We now examine the mean gene expression between inflamed and un-inflamed samples on a patient basis, to see if that showcases a difference in gene expression. The mean gene expression is lower for the un-inflamed patients than for the inflamed patients.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>patient_means <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span> </span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient, inflammation_status) <span class="sc">|&gt;</span> </span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_expr =</span> <span class="fu">mean</span>(gene_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(patient_means, <span class="fu">aes</span>(<span class="at">x =</span> inflammation_status, <span class="at">y =</span> mean_expr)) <span class="sc">+</span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Inflammation status"</span>, <span class="at">y =</span> <span class="st">"Mean gene expression per patient"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/mean_gene_expression_per_patient_inflamed_uninflamed-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"mean_gene_expression_per_patient_inflamed_uninflamed.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="identification-of-significant-genes-during-dependent-on-inflammation-state-using-a-volcano-plot" class="level2">
<h2 class="anchored" data-anchor-id="identification-of-significant-genes-during-dependent-on-inflammation-state-using-a-volcano-plot">Identification of significant genes during dependent on inflammation state using a Volcano plot</h2>
<p>Creating a volcano plot to see which genes are significantly (p &lt; 0.05 ) up-regulated in the un-inflamed and inflamed patients. The biologically relevant genes are selected (the ones with a minimum of log2 fold change of 0.25). The blue dots represent the genes that are significantly up-regulated in un-inflamed patients and the red dots represent the genes that are significantly up-regulated in the inflamed patients. They grey dots represent the genes that did not show any significant up-regulation. The genes that are very significantly up-regulated (p &gt; 10^-10) are labeled.</p>
<p>As can be seen in the plot, there are more genes that are up-regulated in inflamed patients than in non-inflamed patients. Some of the genes that are strongly up-regulated are associated with the immune system, e.g.&nbsp;LCN2 and IGHG1, suggesting increased immune activation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>lfc_cut <span class="ot">&lt;-</span> <span class="fl">0.25</span>   <span class="co"># Minimum |log2FC|</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>p_cut   <span class="ot">&lt;-</span> <span class="fl">0.05</span>   <span class="co"># p-value threshold</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Add gene symbol by matching finaldata$gene_id with featuredata$ID</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>de_results_annot <span class="ot">&lt;-</span> de_results <span class="sc">|&gt;</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">GENE_SYMBOL =</span> featuredata<span class="sc">$</span>GENE_SYMBOL[</span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">match</span>(gene_id, featuredata<span class="sc">$</span>ID)</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Label only highly significant genes with valid symbols</span></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a>label_df <span class="ot">&lt;-</span> de_results_annot <span class="sc">|&gt;</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a>    neg_log10_p <span class="sc">&gt;</span> <span class="dv">10</span>,</span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(GENE_SYMBOL)</span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Prepare plotting data</span></span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a>plot_df <span class="ot">&lt;-</span> de_results_annot <span class="sc">|&gt;</span></span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_value)) <span class="sc">|&gt;</span></span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">p.adjust</span>(p_value, <span class="at">method =</span> <span class="st">"BH"</span>),</span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig_cat =</span> <span class="fu">case_when</span>(</span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a>      FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FC <span class="sc">&gt;=</span>  lfc_cut <span class="sc">~</span> <span class="st">"Up in inflamed"</span>,</span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a>      FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FC <span class="sc">&lt;=</span> <span class="sc">-</span>lfc_cut <span class="sc">~</span> <span class="st">"Up in uninflamed"</span>,</span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span>                            <span class="sc">~</span> <span class="st">"Non significant"</span></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig_cat =</span> <span class="fu">factor</span>(sig_cat,</span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a>                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Up in inflamed"</span>, <span class="st">"Up in uninflamed"</span>, <span class="st">"Non significant"</span>))</span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> plot_df <span class="sc">|&gt;</span></span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">!=</span> <span class="st">"Non significant"</span>) <span class="sc">|&gt;</span></span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, GENE_SYMBOL, log2FC, p_value, FDR, sig_cat) <span class="sc">|&gt;</span></span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR)</span>
<span id="cb101-37"><a href="#cb101-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-38"><a href="#cb101-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Split into up in inflamed / up in uninflamed</span></span>
<span id="cb101-39"><a href="#cb101-39" aria-hidden="true" tabindex="-1"></a>up_in_inflamed <span class="ot">&lt;-</span> sig_genes <span class="sc">|&gt;</span></span>
<span id="cb101-40"><a href="#cb101-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">==</span> <span class="st">"Up in inflamed"</span>) <span class="sc">|&gt;</span></span>
<span id="cb101-41"><a href="#cb101-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb101-42"><a href="#cb101-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-43"><a href="#cb101-43" aria-hidden="true" tabindex="-1"></a>up_in_uninflamed <span class="ot">&lt;-</span> sig_genes <span class="sc">|&gt;</span></span>
<span id="cb101-44"><a href="#cb101-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">==</span> <span class="st">"Up in uninflamed"</span>) <span class="sc">|&gt;</span></span>
<span id="cb101-45"><a href="#cb101-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb101-46"><a href="#cb101-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-47"><a href="#cb101-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb101-48"><a href="#cb101-48" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_df,</span>
<span id="cb101-49"><a href="#cb101-49" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FC, <span class="at">y =</span> neg_log10_p, <span class="at">colour =</span> sig_cat)) <span class="sc">+</span></span>
<span id="cb101-50"><a href="#cb101-50" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Points</span></span>
<span id="cb101-51"><a href="#cb101-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb101-52"><a href="#cb101-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Cutoff lines</span></span>
<span id="cb101-53"><a href="#cb101-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(p_cut),</span>
<span id="cb101-54"><a href="#cb101-54" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb101-55"><a href="#cb101-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>lfc_cut, lfc_cut),</span>
<span id="cb101-56"><a href="#cb101-56" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb101-57"><a href="#cb101-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># labels</span></span>
<span id="cb101-58"><a href="#cb101-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb101-59"><a href="#cb101-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> label_df,</span>
<span id="cb101-60"><a href="#cb101-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> GENE_SYMBOL),</span>
<span id="cb101-61"><a href="#cb101-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb101-62"><a href="#cb101-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"black"</span>,</span>
<span id="cb101-63"><a href="#cb101-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="dv">30</span>,</span>
<span id="cb101-64"><a href="#cb101-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fl">0.4</span>,</span>
<span id="cb101-65"><a href="#cb101-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">point.padding =</span> <span class="fl">0.3</span>,</span>
<span id="cb101-66"><a href="#cb101-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">segment.size =</span> <span class="fl">0.2</span>,</span>
<span id="cb101-67"><a href="#cb101-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.legend =</span> <span class="cn">FALSE</span></span>
<span id="cb101-68"><a href="#cb101-68" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb101-69"><a href="#cb101-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-70"><a href="#cb101-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(</span>
<span id="cb101-71"><a href="#cb101-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb101-72"><a href="#cb101-72" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in inflamed"</span>   <span class="ot">=</span> <span class="st">"#d73027"</span>,</span>
<span id="cb101-73"><a href="#cb101-73" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in uninflamed"</span> <span class="ot">=</span> <span class="st">"#4575b4"</span>,</span>
<span id="cb101-74"><a href="#cb101-74" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Non significant"</span>               <span class="ot">=</span> <span class="st">"grey80"</span></span>
<span id="cb101-75"><a href="#cb101-75" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb101-76"><a href="#cb101-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="cn">NULL</span></span>
<span id="cb101-77"><a href="#cb101-77" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb101-78"><a href="#cb101-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb101-79"><a href="#cb101-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Volcano plot of colonic gene expression"</span>,</span>
<span id="cb101-80"><a href="#cb101-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Inflamed vs uninflamed biopsies (Gut 57:1398–1405)"</span>,</span>
<span id="cb101-81"><a href="#cb101-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"log2 fold change (Inflamed – Uninflamed)"</span>,</span>
<span id="cb101-82"><a href="#cb101-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)<span class="sc">~</span>value))</span>
<span id="cb101-83"><a href="#cb101-83" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb101-84"><a href="#cb101-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb101-85"><a href="#cb101-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb101-86"><a href="#cb101-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb101-87"><a href="#cb101-87" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb101-88"><a href="#cb101-88" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb101-89"><a href="#cb101-89" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb101-90"><a href="#cb101-90" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb101-91"><a href="#cb101-91" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/volcano_plot_colonic_gene_expression-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"volcano_plot_colonic_gene_expression.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="expression-of-the-40-most-significant-genes" class="level2">
<h2 class="anchored" data-anchor-id="expression-of-the-40-most-significant-genes">Expression of the 40 most significant genes</h2>
<p>Plotting the expression of the 40 most significant genes in both inflamed and un-inflamed patients. The inflamed patients have a higher expression of some of the genes associated with immune activation compared to the un-inflamed patients. This is for example LCN2 and IGHG1 which is in compliance with the results from the volcano plot. Some of the genes that are down-regulated include GCNT2 and UGT2A3 which are housekeeping genes that are suppressed to free up resources.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Pick top 40 most significant genes</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> sig_genes <span class="sc">|&gt;</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(FDR, <span class="at">n =</span> <span class="dv">40</span>) <span class="sc">|&gt;</span></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, GENE_SYMBOL)</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Join</span></span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>heat_df <span class="ot">&lt;-</span> finaldata <span class="sc">|&gt;</span></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(top_markers, <span class="at">by =</span> <span class="st">"gene_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GENE_SYMBOL)) <span class="sc">|&gt;</span></span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(patient, inflammation_status, GENE_SYMBOL, gene_expr) <span class="sc">|&gt;</span></span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient =</span> <span class="fu">factor</span>(patient))     <span class="co">#Make patient discrete</span></span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Reorder symbols by average expression difference</span></span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a>symbol_order <span class="ot">&lt;-</span> heat_df <span class="sc">|&gt;</span></span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENE_SYMBOL) <span class="sc">|&gt;</span></span>
<span id="cb103-16"><a href="#cb103-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_expr =</span> <span class="fu">mean</span>(gene_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">|&gt;</span></span>
<span id="cb103-17"><a href="#cb103-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mean_expr)) <span class="sc">|&gt;</span></span>
<span id="cb103-18"><a href="#cb103-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb103-19"><a href="#cb103-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-20"><a href="#cb103-20" aria-hidden="true" tabindex="-1"></a>heat_df<span class="sc">$</span>GENE_SYMBOL <span class="ot">&lt;-</span> <span class="fu">factor</span>(heat_df<span class="sc">$</span>GENE_SYMBOL, <span class="at">levels =</span> symbol_order)</span>
<span id="cb103-21"><a href="#cb103-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-22"><a href="#cb103-22" aria-hidden="true" tabindex="-1"></a><span class="co"># draw heatmap</span></span>
<span id="cb103-23"><a href="#cb103-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(heat_df,</span>
<span id="cb103-24"><a href="#cb103-24" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> patient, <span class="at">y =</span> GENE_SYMBOL, <span class="at">fill =</span> gene_expr)) <span class="sc">+</span></span>
<span id="cb103-25"><a href="#cb103-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb103-26"><a href="#cb103-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb103-27"><a href="#cb103-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> inflammation_status, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb103-28"><a href="#cb103-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb103-29"><a href="#cb103-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Patient"</span>,</span>
<span id="cb103-30"><a href="#cb103-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Gene symbol"</span>,</span>
<span id="cb103-31"><a href="#cb103-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Expression"</span>,</span>
<span id="cb103-32"><a href="#cb103-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Top 40 differential genes across samples"</span></span>
<span id="cb103-33"><a href="#cb103-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb103-34"><a href="#cb103-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb103-35"><a href="#cb103-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb103-36"><a href="#cb103-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x  =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-37"><a href="#cb103-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-38"><a href="#cb103-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid   =</span> <span class="fu">element_blank</span>()</span>
<span id="cb103-39"><a href="#cb103-39" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/top40_differential_genes_across_samples-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"top40_differential_genes_across_samples.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="correlation-between-time-untreated-and-gene-expression" class="level2">
<h2 class="anchored" data-anchor-id="correlation-between-time-untreated-and-gene-expression">Correlation between time untreated and gene expression</h2>
<p>Plotting the correlation between the time patients went untreated and the gene expression of significant genes. Looking at the genes that were up-regulated in inflamed patients (red) it can be seen that some of the patients with the genes that were the most up-regulated went to the doctor earlier compared to some of the patients that had genes that were the least up-regulated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>finaldata <span class="sc">|&gt;</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_id <span class="sc">==</span> sig_genes<span class="sc">$</span>gene_id) <span class="sc">|&gt;</span> </span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geo_accession, gene_id, time_untreated, disease, family_history) <span class="sc">|&gt;</span> </span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sig_genes, finaldata, <span class="at">by =</span> <span class="st">"gene_id"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENE_SYMBOL) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">|&gt;</span> </span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time_untreated <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time_untreated, <span class="at">y =</span> log2FC, </span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> sig_cat)) <span class="sc">+</span></span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Correlation between time untreated and expression of significant genes"</span>, <span class="at">x =</span> <span class="st">"Time untreated (days)"</span>, <span class="at">y =</span> <span class="st">"log2FC"</span>, <span class="at">color =</span> <span class="st">"Inflammation status"</span>) <span class="sc">+</span> </span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `filter()`.
ℹ In argument: `gene_id == sig_genes$gene_id`.
Caused by warning in `gene_id == sig_genes$gene_id`:
! longer object length is not a multiple of shorter object length</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/correlation_time_untreated_significant_genes-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"correlation_time_untreated_significant_genes.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-environment-4" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment-4">Cleaning environment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="forrest-modeling" class="level1">
<h1>Forrest modeling</h1>
<section id="loading-libraries-7" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-7">Loading libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(randomForest)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">6969</span>) <span class="co">#Setting seed for reproducability</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>expr_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_rf_expr_df.csv.gz"</span>) <span class="co">#used for input nodes</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>featuredata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_dat_genbank.csv.gz"</span>) <span class="co">#used for classifier target</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_de_results.csv.gz"</span>) <span class="co">#required for vulcano plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="model-training" class="level2">
<h2 class="anchored" data-anchor-id="model-training">Model Training</h2>
<section id="splitting-dataset-into-train-and-test" class="level3">
<h3 class="anchored" data-anchor-id="splitting-dataset-into-train-and-test">Splitting dataset into train and test</h3>
<p>We start by splitting the data into train and test data sets. The model is allowed to “learn” correlations on train, and it’s performance can then be evaluated using test_data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>percent_train <span class="ot">&lt;-</span> <span class="fl">0.8</span> <span class="co">#use 80% of the data for train </span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>train_index <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(expr_df), <span class="fl">0.8</span> <span class="sc">*</span> <span class="fu">nrow</span>(expr_df))</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> expr_df[train_index, ]</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>test_data <span class="ot">&lt;-</span> expr_df[<span class="sc">-</span>train_index, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Its important that the model never “sees” the test data, as we need the model performance metrics to be truly independent. This reduces the chance of the model fitting to correlations specific to the data values, rather than a biological signal.</p>
</section>
<section id="running-the-rf-classifier-model" class="level3">
<h3 class="anchored" data-anchor-id="running-the-rf-classifier-model">Running the RF Classifier model</h3>
<p>The target value “y_pred_UC” for the model is whether the sample was from an individual with (1) or without (0) UC. This column is prepared previously in 03_augment.qmd.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co">#y_pred_UC as factor was not implemented in 03_augment, despite the use of as.factor()</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> train_data <span class="sc">|&gt;</span> </span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_pred_UC =</span> <span class="fu">as.factor</span>(y_pred_UC))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Each gene is its own input node for the model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># y_pred_UC ~ . </span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>train_data.rf <span class="ot">&lt;-</span> <span class="fu">randomForest</span>(y_pred_UC <span class="sc">~</span> . <span class="sc">-</span> geo_accession, </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">type =</span> <span class="st">"classification"</span>,</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> train_data, </span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">mtry =</span> <span class="dv">3</span>,</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">importance =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a><span class="co"># importance = True is used to obtain most contributing inputs; aka genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="model-evaluation" class="level2">
<h2 class="anchored" data-anchor-id="model-evaluation">Model Evaluation</h2>
<section id="deriving-the-confusion-matrix-of-the-rf-model" class="level3">
<h3 class="anchored" data-anchor-id="deriving-the-confusion-matrix-of-the-rf-model">Deriving the Confusion Matrix of the RF model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>test_pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(train_data.rf, test_data)</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>confusion_matrix <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">Predicted =</span> test_pred, <span class="at">Actual =</span> test_data<span class="sc">$</span>y_pred_UC)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(confusion_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         Actual
Predicted  0  1
        0  6  6
        1  3 26</code></pre>
</div>
</div>
<p>Confusion matrix shows the amount of times the model predicted 0 or 1 (rows), based on the actual values of 0 or 1 in the test dataset. It therefore shows:</p>
<table class="caption-top table">
<tbody>
<tr class="odd">
<td style="text-align: center;">TN</td>
<td style="text-align: center;">FN</td>
</tr>
<tr class="even">
<td style="text-align: center;">FP</td>
<td style="text-align: center;">TP</td>
</tr>
</tbody>
</table>
<p>where;</p>
<ul>
<li><p>TN: True Negative</p></li>
<li><p>FN: False Negative</p></li>
<li><p>FP: False Positive</p></li>
<li><p>TP: True Positive</p></li>
</ul>
<p>From the matrix, we can obtain an accuracy score.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Correct predictions; TN, TP</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>TN <span class="ot">&lt;-</span> confusion_matrix[<span class="dv">1</span>, <span class="dv">1</span>]</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>TP <span class="ot">&lt;-</span> confusion_matrix[<span class="dv">2</span>, <span class="dv">2</span>]</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>FP <span class="ot">&lt;-</span> confusion_matrix[<span class="dv">2</span>, <span class="dv">1</span>]</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>FN <span class="ot">&lt;-</span> confusion_matrix[<span class="dv">1</span>, <span class="dv">2</span>]</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> <span class="fu">sum</span>(confusion_matrix)</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>acc <span class="ot">&lt;-</span> (TN<span class="sc">+</span>TP)<span class="sc">/</span>total</span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(acc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.7804878</code></pre>
</div>
</div>
<p>The model emitted a 78% accuracy.</p>
</section>
<section id="importance-measure-of-genes" class="level3">
<h3 class="anchored" data-anchor-id="importance-measure-of-genes">Importance measure of Genes</h3>
<p>We would like to find the model contributing genes in the model, to see which genes (at least in the eyes of the model), speaks to samples with and without UC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>var_importance <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">round</span>(<span class="fu">importance</span>(train_data.rf), <span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>var_importance <span class="ot">&lt;-</span> var_importance <span class="sc">|&gt;</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">str_split_i</span>(<span class="fu">rownames</span>(var_importance), <span class="st">"_"</span>, <span class="dv">2</span>)) <span class="sc">|&gt;</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">GENE_SYMBOL =</span> featuredata<span class="sc">$</span>GENE_SYMBOL[</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">match</span>(gene_id, featuredata<span class="sc">$</span>ID)</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>var_importance <span class="sc">|&gt;</span> </span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(gene_id = str_split_i(rownames(var_importance), "_", 2)) |&gt; </span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(MeanDecreaseAccuracy)) <span class="sc">|&gt;</span> </span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>) <span class="sc">|&gt;</span></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use reorder() to automatically sort the bars by the MeanDecreaseAccuracy value</span></span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">reorder</span>(GENE_SYMBOL, MeanDecreaseAccuracy), </span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> MeanDecreaseAccuracy</span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb120-10"><a href="#cb120-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb120-11"><a href="#cb120-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb120-12"><a href="#cb120-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb120-13"><a href="#cb120-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Top 12 Most Contributing Genes in RF"</span>,</span>
<span id="cb120-14"><a href="#cb120-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Gene"</span>,</span>
<span id="cb120-15"><a href="#cb120-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Mean decrease accuracy from RF"</span></span>
<span id="cb120-16"><a href="#cb120-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb120-17"><a href="#cb120-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-130-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"top15_most_contributing_genes_RF.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Extract gene_ids as list</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>most_contributing_vars <span class="ot">&lt;-</span> var_importance <span class="sc">|&gt;</span> </span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(MeanDecreaseAccuracy)) <span class="sc">|&gt;</span> </span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>) <span class="sc">|&gt;</span> </span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="match-gene_id-with-gene_symbol-to-match-other-analysis" class="level3">
<h3 class="anchored" data-anchor-id="match-gene_id-with-gene_symbol-to-match-other-analysis">Match gene_id with gene_symbol to match other analysis</h3>
<p>GENE_SYMBOL can be found in featuredata. This is used to map to the results of the vulcano plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Splitting gene_id into only the number and converting to numeric</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>most_contributing_vars <span class="ot">&lt;-</span> most_contributing_vars <span class="sc">|&gt;</span> </span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_split_i</span>(<span class="st">"_"</span>, <span class="dv">2</span>) <span class="sc">|&gt;</span> </span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">parse_number</span>()</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>most_contributing_vars <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(most_contributing_vars)</span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>most_contributing_vars <span class="ot">&lt;-</span> most_contributing_vars <span class="sc">|&gt;</span> </span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> most_contributing_vars) <span class="sc">|&gt;</span> </span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>most_contributing_gene_symbols <span class="ot">&lt;-</span> featuredata <span class="sc">|&gt;</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID, GENE_SYMBOL) <span class="sc">|&gt;</span> </span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(most_contributing_vars, featuredata, <span class="at">by=</span><span class="st">"ID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(GENE_SYMBOL) <span class="sc">|&gt;</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GENE_SYMBOL)) <span class="sc">|&gt;</span> </span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(most_contributing_gene_symbols)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "CDH24"    "S100A5"   "PLB1"     "PREI3"    "EGR1"     "DEFA5"   
 [7] "C8orf46"  "KIAA0484" "IGFBP2"   "RHBDL1"   "RSNL2"    "PTGDS"   </code></pre>
</div>
</div>
</section>
</section>
<section id="comparison-of-identified-genes-impacting-inflammation-with-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="comparison-of-identified-genes-impacting-inflammation-with-disease-state">Comparison of identified genes impacting inflammation with disease state</h2>
<p>To visualize overlay between significant genes found in 05_analysis, and the RandomForest model most contributing variables.</p>
<p>The 05_analysis relied on inflammation status, and the model relied on disease state.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/03_de_results.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>lfc_cut <span class="ot">&lt;-</span> <span class="fl">0.25</span>   <span class="co"># minimum |log2FC|</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>p_cut   <span class="ot">&lt;-</span> <span class="fl">0.05</span>   <span class="co"># p-value threshold for the dashed line</span></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbol by matching finaldata$gene_id --&gt; featuredata$ID</span></span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a>de_results_annot <span class="ot">&lt;-</span> de_results <span class="sc">|&gt;</span></span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">GENE_SYMBOL =</span> featuredata<span class="sc">$</span>GENE_SYMBOL[</span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">match</span>(gene_id, featuredata<span class="sc">$</span>ID)</span>
<span id="cb127-9"><a href="#cb127-9" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb127-10"><a href="#cb127-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb127-11"><a href="#cb127-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-12"><a href="#cb127-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Label only highly significant genes with valid symbols</span></span>
<span id="cb127-13"><a href="#cb127-13" aria-hidden="true" tabindex="-1"></a>label_df <span class="ot">&lt;-</span> de_results_annot <span class="sc">|&gt;</span></span>
<span id="cb127-14"><a href="#cb127-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb127-15"><a href="#cb127-15" aria-hidden="true" tabindex="-1"></a>    neg_log10_p <span class="sc">&gt;</span> <span class="dv">10</span>,         </span>
<span id="cb127-16"><a href="#cb127-16" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(GENE_SYMBOL)</span>
<span id="cb127-17"><a href="#cb127-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb127-18"><a href="#cb127-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-19"><a href="#cb127-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting data</span></span>
<span id="cb127-20"><a href="#cb127-20" aria-hidden="true" tabindex="-1"></a>plot_df <span class="ot">&lt;-</span> de_results_annot <span class="sc">|&gt;</span></span>
<span id="cb127-21"><a href="#cb127-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_value)) <span class="sc">|&gt;</span></span>
<span id="cb127-22"><a href="#cb127-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb127-23"><a href="#cb127-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">p.adjust</span>(p_value, <span class="at">method =</span> <span class="st">"BH"</span>),</span>
<span id="cb127-24"><a href="#cb127-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig_cat =</span> <span class="fu">case_when</span>(</span>
<span id="cb127-25"><a href="#cb127-25" aria-hidden="true" tabindex="-1"></a>      FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FC <span class="sc">&gt;=</span>  lfc_cut <span class="sc">~</span> <span class="st">"Up in inflamed"</span>,</span>
<span id="cb127-26"><a href="#cb127-26" aria-hidden="true" tabindex="-1"></a>      FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FC <span class="sc">&lt;=</span> <span class="sc">-</span>lfc_cut <span class="sc">~</span> <span class="st">"Up in uninflamed"</span>,</span>
<span id="cb127-27"><a href="#cb127-27" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span>                           <span class="sc">~</span> <span class="st">"Non significant"</span></span>
<span id="cb127-28"><a href="#cb127-28" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb127-29"><a href="#cb127-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig_cat =</span> <span class="fu">factor</span>(sig_cat,</span>
<span id="cb127-30"><a href="#cb127-30" aria-hidden="true" tabindex="-1"></a>                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Up in inflamed"</span>, <span class="st">"Up in uninflamed"</span>, <span class="st">"Non significant"</span>)),</span>
<span id="cb127-31"><a href="#cb127-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb127-32"><a href="#cb127-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Identify genes from the highlight list</span></span>
<span id="cb127-33"><a href="#cb127-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">highlight_cat =</span> <span class="fu">case_when</span>(</span>
<span id="cb127-34"><a href="#cb127-34" aria-hidden="true" tabindex="-1"></a>      GENE_SYMBOL <span class="sc">%in%</span> most_contributing_gene_symbols <span class="sc">~</span> <span class="st">"Most Contributing"</span>,</span>
<span id="cb127-35"><a href="#cb127-35" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Other"</span></span>
<span id="cb127-36"><a href="#cb127-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb127-37"><a href="#cb127-37" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb127-38"><a href="#cb127-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-39"><a href="#cb127-39" aria-hidden="true" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> plot_df <span class="sc">|&gt;</span></span>
<span id="cb127-40"><a href="#cb127-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">!=</span> <span class="st">"Non significant"</span>) <span class="sc">|&gt;</span></span>
<span id="cb127-41"><a href="#cb127-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, GENE_SYMBOL, log2FC, p_value, FDR, sig_cat) <span class="sc">|&gt;</span></span>
<span id="cb127-42"><a href="#cb127-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR)</span>
<span id="cb127-43"><a href="#cb127-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-44"><a href="#cb127-44" aria-hidden="true" tabindex="-1"></a><span class="co"># split into up in inflamed / up in uninflamed</span></span>
<span id="cb127-45"><a href="#cb127-45" aria-hidden="true" tabindex="-1"></a>up_in_inflamed <span class="ot">&lt;-</span> sig_genes <span class="sc">|&gt;</span></span>
<span id="cb127-46"><a href="#cb127-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">==</span> <span class="st">"Up in inflamed"</span>) <span class="sc">|&gt;</span></span>
<span id="cb127-47"><a href="#cb127-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb127-48"><a href="#cb127-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-49"><a href="#cb127-49" aria-hidden="true" tabindex="-1"></a>up_in_uninflamed <span class="ot">&lt;-</span> sig_genes <span class="sc">|&gt;</span></span>
<span id="cb127-50"><a href="#cb127-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sig_cat <span class="sc">==</span> <span class="st">"Up in uninflamed"</span>) <span class="sc">|&gt;</span></span>
<span id="cb127-51"><a href="#cb127-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GENE_SYMBOL)</span>
<span id="cb127-52"><a href="#cb127-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-53"><a href="#cb127-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano plot with Highlighted Genes</span></span>
<span id="cb127-54"><a href="#cb127-54" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_df,</span>
<span id="cb127-55"><a href="#cb127-55" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FC, <span class="at">y =</span> neg_log10_p)) <span class="sc">+</span></span>
<span id="cb127-56"><a href="#cb127-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-57"><a href="#cb127-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 1. Base points (coloured by sig_cat)</span></span>
<span id="cb127-58"><a href="#cb127-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> sig_cat), <span class="at">size =</span> <span class="fl">1.3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb127-59"><a href="#cb127-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-60"><a href="#cb127-60" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 2. Overlay highlighted genes (colored by highlight_cat)</span></span>
<span id="cb127-61"><a href="#cb127-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">filter</span>(plot_df, highlight_cat <span class="sc">==</span> <span class="st">"Most Contributing"</span>),</span>
<span id="cb127-62"><a href="#cb127-62" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">color =</span> highlight_cat),</span>
<span id="cb127-63"><a href="#cb127-63" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="fl">1.3</span>) <span class="sc">+</span></span>
<span id="cb127-64"><a href="#cb127-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-65"><a href="#cb127-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># cutoff lines</span></span>
<span id="cb127-66"><a href="#cb127-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(p_cut),</span>
<span id="cb127-67"><a href="#cb127-67" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb127-68"><a href="#cb127-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>lfc_cut, lfc_cut),</span>
<span id="cb127-69"><a href="#cb127-69" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb127-70"><a href="#cb127-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-71"><a href="#cb127-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># labels for strongest hits</span></span>
<span id="cb127-72"><a href="#cb127-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb127-73"><a href="#cb127-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> label_df,</span>
<span id="cb127-74"><a href="#cb127-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> GENE_SYMBOL),</span>
<span id="cb127-75"><a href="#cb127-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb127-76"><a href="#cb127-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"black"</span>,</span>
<span id="cb127-77"><a href="#cb127-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="dv">30</span>,</span>
<span id="cb127-78"><a href="#cb127-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fl">0.4</span>,</span>
<span id="cb127-79"><a href="#cb127-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">point.padding =</span> <span class="fl">0.3</span>,</span>
<span id="cb127-80"><a href="#cb127-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">segment.size =</span> <span class="fl">0.2</span>,</span>
<span id="cb127-81"><a href="#cb127-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.legend =</span> <span class="cn">FALSE</span></span>
<span id="cb127-82"><a href="#cb127-82" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb127-83"><a href="#cb127-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-84"><a href="#cb127-84" aria-hidden="true" tabindex="-1"></a>  <span class="co"># unified color scale (sig categories + highlighted category)</span></span>
<span id="cb127-85"><a href="#cb127-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb127-86"><a href="#cb127-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb127-87"><a href="#cb127-87" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in inflamed"</span>   <span class="ot">=</span> <span class="st">"#d73027"</span>,</span>
<span id="cb127-88"><a href="#cb127-88" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in uninflamed"</span> <span class="ot">=</span> <span class="st">"#4575b4"</span>,</span>
<span id="cb127-89"><a href="#cb127-89" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Non significant"</span>  <span class="ot">=</span> <span class="st">"grey80"</span>,</span>
<span id="cb127-90"><a href="#cb127-90" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Most Contributing"</span> <span class="ot">=</span> <span class="st">"green"</span></span>
<span id="cb127-91"><a href="#cb127-91" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb127-92"><a href="#cb127-92" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="cn">NULL</span>,</span>
<span id="cb127-93"><a href="#cb127-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb127-94"><a href="#cb127-94" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in inflamed"</span>   <span class="ot">=</span> <span class="st">"Up in inflamed"</span>,</span>
<span id="cb127-95"><a href="#cb127-95" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Up in uninflamed"</span> <span class="ot">=</span> <span class="st">"Up in uninflamed"</span>,</span>
<span id="cb127-96"><a href="#cb127-96" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Non significant"</span>  <span class="ot">=</span> <span class="st">"Non significant"</span>,</span>
<span id="cb127-97"><a href="#cb127-97" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Most Contributing"</span> <span class="ot">=</span> <span class="st">"Genes affecting UC diagnosis present in RF model"</span></span>
<span id="cb127-98"><a href="#cb127-98" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb127-99"><a href="#cb127-99" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb127-100"><a href="#cb127-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-101"><a href="#cb127-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb127-102"><a href="#cb127-102" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Volcano plot of colonic gene expression"</span>,</span>
<span id="cb127-103"><a href="#cb127-103" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Inflamed vs uninflamed biopsies (Gut 57:1398–1405)"</span>,</span>
<span id="cb127-104"><a href="#cb127-104" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"log2 fold change (Inflamed – Uninflamed)"</span>,</span>
<span id="cb127-105"><a href="#cb127-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)<span class="sc">~</span>value))</span>
<span id="cb127-106"><a href="#cb127-106" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb127-107"><a href="#cb127-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb127-108"><a href="#cb127-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb127-109"><a href="#cb127-109" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb127-110"><a href="#cb127-110" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb127-111"><a href="#cb127-111" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb127-112"><a href="#cb127-112" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb127-113"><a href="#cb127-113" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb127-114"><a href="#cb127-114" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-135-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"volcano_plot_colonic_gene_expression_RF.png"</span>, <span class="at">path =</span> <span class="st">"../results"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-environment-5" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-environment-5">Cleaning environment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>