---
title: "Presentation"
author: "Asbj√∏rn, Emiliano, Magnus, Oliver and Trine"
format:
  revealjs:
    theme: simple
    transition: slide
    footer: "Group 01 R for Bio data science 22160"
    highlight-style: github
    code-fold: false
    slide-number: c/t
    show-slide-number: all
output-dir: doc
---

# Introduction 1

::: {#Introduction style="font-size: 0.8em"}
-   Dataset from NCBI

-   Transcriptional profiling of colon epithelial from ulcerative colitis patients and healthy controls

-   Ulcers in the colon

-   chronic inflamation

-   Aim to find and analyze variation between healthy controls and patients

    -   what smoking does
    -   Effect of genes
:::

# Materials of Methods 2

::::: columns
::: {.column width="50%" style="font-size: 0.8em"}
-   Biopsies from the gut
    -   202 samples

    -   from 67 patients

    -   30 healthy controls
-   Gene expression
-   Augtments to the data
    -   Age and Age group
    -   Time untreated
:::

::: {.column width="50%"}
![](images/age_groups_distribution_inflammation_status.png)
:::
:::::

::: {#Source .Reference style="font-size: 0.4em"}
*Noble CL, Abbas AR, Cornelius J, et al Regional variation in gene expression in the healthy colon is dysregulated in ulcerative colitis Gut 2008;**57:**1398-1405.*
:::

## Filtering data 3

::::: columns
::: {.column width="50%" style="font-size: 0.9em"}
Removing one-type columns

```{r Cleaning_phenodata, echo=TRUE, eval=FALSE}
phenodata <- phenodata |>
  select(
    where(
      ~ n_distinct(
        .x, na.rm = TRUE) > 1))
```

![](images/variance_distribution_across_genes.png)
:::

::: {.column width="50%" style="font-size: 0.9em"}
Removing low variance genes - 37000 of 40000 leaving 10 %

```{r Cleaning_genes, echo=TRUE, eval=FALSE}
gene_var <- exprs_df |> 
  # Drop the sample ID column, keep only expression values
  select(-geo_accession) |> 
  summarise(across(everything(), ~ var(.x, na.rm = TRUE))) |> 
  pivot_longer(
    cols      = everything(),
    names_to  = "gene",
    values_to = "variance"
  )
# Cutoff genes that does not vary more than 0.05 - 2909 genes remaining
genes_keep <- gene_var |> 
  filter(variance > 0.05) |> 
  pull(gene)

exprs_filtered <- exprs_df |> 
  select(geo_accession, all_of(genes_keep))

```
:::
:::::

# Results 4

::::: columns
::: {.column width="40%"}
-   Key points

![](images/confusion_matrix_rf.png)
:::

::: {.column width="60%"}
![](images/volcano_plot_colonic_gene_expression_RF-01.png)
:::
:::::

# plot 5

::::: columns
::: {.column width="30%"}
-   key points
:::

::: {.column width="70%"}
![](images/top40_differential_genes_across_samples.png)
:::
:::::

# Plot 6

::::: columns
::: {.column width="30%"}
-   key points
:::

::: {.column width="70%"}
![](images/significant_gene_count_by_anatomic_location.png)
:::
:::::

# Discussion 8

::: {#Discussion style="font-size: 0.8em"}
-   What causes UC

    -   We investigated if there is a genetic bias in getting UC

    -   We investigated the role of smoking

-   Limited sample size (67 patients + 30 controls)

-   Could be caused by

    -   Immune-system issue i.e. an auto immune diesise

    -   Gut bacteria imbalance

    -   Genetic traits

    -   Environmental impact

-   An inconsistent data set - Meta data
:::

::: {#Source2 style="font-size: 0.4em"}
<https://www.mayoclinic.org/diseases-conditions/ulcerative-colitis/symptoms-causes/syc-20353326>
:::

# Conclusion 9

::: {#Conclusion style="font-size: 0.8em"}
-   Correlation and causality

We showed a Correlation between some genes and ulcerative colitis (UC). However this does not show us if it is caused by the UC or the cause of the UC.

We developed a randomforest model which predicts the significance of genes in UC so this could be used either as a diagnostic tool or for prediction of a future UC patient, however to test this model we would need more data. probably do a genome wide association study

We did not find a correlation between UC and smoking nor a correlation between UC and age. Proving it can become an issue for all people
:::
